<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc" />


<title>EigenMS</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/paper.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Thesis BN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="regression_thesis.html">My preprocess tests</a>
</li>
<li>
  <a href="models_tests.html">Different models tests</a>
</li>
<li>
  <a href="bn_tests.html">BN tests</a>
</li>
<li>
  <a href="visualizations.html">Visualizations</a>
</li>
<li>
  <a href="waveica2.html">WaveICA2.0</a>
</li>
<li>
  <a href="combat.html">ComBat</a>
</li>
<li>
  <a href="pqn.html">PQN</a>
</li>
<li>
  <a href="eigenms.html">EigenMS</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/marcsole96/BN_thesis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">EigenMS</h1>
<h4 class="author">Marc</h4>
<h4 class="date">30/3/2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-04
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>workflowr/data/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220214code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220214)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220214code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220214)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/mysit/Documents/Bioinformatica/Semestre_4/workflowr/data
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommarcsole96BNthesistree30da757975fdf43d2b9016f035970f6047e3c02ftargetblank30da757a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/marcsole96/BN_thesis/tree/30da757975fdf43d2b9016f035970f6047e3c02f" target="_blank">30da757</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommarcsole96BNthesistree30da757975fdf43d2b9016f035970f6047e3c02ftargetblank30da757a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/marcsole96/BN_thesis/tree/30da757975fdf43d2b9016f035970f6047e3c02f" target="_blank">30da757</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/preprocessing_thesis_cache/
    Ignored:    data/Demo for WaveICA/

Untracked files:
    Untracked:  data/.Rprofile
    Untracked:  data/.gitignore
    Untracked:  data/NOREVA.2.1.1.tar.gz
    Untracked:  data/TraceAge_bloodspots_t3_neg_clean.csv
    Untracked:  data/TraceAge_bloodspots_t3_pos_clean.csv
    Untracked:  data/bn_after_norm.rds
    Untracked:  data/bn_before_norm.rds
    Untracked:  data/bn_means.rds
    Untracked:  data/bn_test.rds
    Untracked:  data/combat_model.rds
    Untracked:  data/data_for_BN.RData
    Untracked:  data/mdf.rds
    Untracked:  data/mdf_processed.rds
    Untracked:  data/model_tests.rds
    Untracked:  data/models.RData
    Untracked:  data/models_waveICA2.0.RData
    Untracked:  data/packrat/
    Untracked:  data/pqn_msbox.rds
    Untracked:  data/pqn_rcpm.rds
    Untracked:  data/sample_order_traceage_wp2_sample_overview.xlsx
    Untracked:  data/test_peptides.txt
    Untracked:  data/waveICA_models.rds
    Untracked:  models.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/eigenms.Rmd</code>) and HTML (<code>docs/eigenms.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/marcsole96/BN_thesis/f0344694a70d4ce057c080a5803a09154365a19c/docs/eigenms.html" target="_blank">f034469</a>
</td>
<td>
marcsole96
</td>
<td>
2022-04-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/marcsole96/BN_thesis/blob/9b245074d8344df5120a090507dcbfdb153574af/analysis/eigenms.Rmd" target="_blank">9b24507</a>
</td>
<td>
marcsole96
</td>
<td>
2022-04-04
</td>
<td>
test
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>#EigenMS</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># This software is released unde The GNU General Public License:</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># http://www.gnu.org/licenses/gpl.html</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># EigenMS normalization</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Ref: &quot;Normalization of peak intensities in bottom-up MS-based proteomics using</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#       singular value decomposition&quot; Karpievitch YV, Taverner T, Adkins JN,</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#       Callister SJ, Anderson GA, Smith RD, Dabney AR. Bioinformatics 2009</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># and</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Ref: &quot;Metabolomics data normalization with EigenMS&quot;</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="co">#       Karpievitch YK, Nikolic SB, Wilson R, Sharman JE, Edwards LM</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#       Submitted to PLoS ONE</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Here we allow multiple facotrs to be preserved in ANOVA model before identifying bais. </span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co"># This requires a bit of thought on the side of the researchers, only few factors should be &#39;preserved&#39;.</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co"># For example &#39;treatment group&#39; is important to preserve but &#39;age&#39; may or may not be important to preserve.</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Here we do not utilize peptides with 1+ grp missing completely, this is a separate problem </span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="co"># addressed by Wang et al, 2011</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Written by Yuliya Karpievitch, Tom Taverner, and Shelley Herbrich </span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># email: yuliya.k@gmail.com</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Version has been split into 2 main functions: eig_norm1 finds significant bias trends,</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co"># then the user can decide if he/she wants to use that number.</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="co"># For Matabolomics data we suggest useing 20% of the number of samples examined </span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co"># as the number of bias trends to eliminated. By setting </span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># ints_eig1$h.c = ceil(.2 * num_samples)</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co"># if the number of automatically identified bias trends are close to that number we suggest </span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co"># using the estimated number of bias trends.</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co"># eig_norm2 function normalizes the data</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that rescaling has been abandoned in the latest version due to discussions abot the fact </span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co"># that systematic bias is what has been inadvertently added, thus we need to remove the bias </span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="co"># but nto add any additional variation. </span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Metabolomics in particular has a lot of variation that still remains that we concluded </span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co"># there is not need for rescaling. </span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="co"># EigenMS estimates and preserves fixed effects. </span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="co"># Contributions to using Random effects are welcome. </span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="co"># HOWTO RUN: </span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co"># source(&#39;~/EigenMS/EigenMS.R&#39;)</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co"># dd = # read in the data</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="co"># grps = # read in group information file </span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co"># logInts = # subset the dd matrix to only hte portion that contains intensities</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co">#           # replace 0&#39;s with NAs, do a log-transformation to approximate Normality.</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="co"># prot.info = cbind(data.frame(rownames(dd)), data.frame(rownames(dd))) # peptideIDs, no PR IDs here, just duplicate column</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co">#             # in case od protein IDs, those are not importnat for normalization and will be ignored.  </span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co"># ints_eig1 = eig_norm1(m=logInts, treatment=grps, prot.info=prot.info)</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co"># ints_norm = eig_norm2(rv=ints_eig1) </span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="co"># eig_norm1 = function(m, treatment, prot.info, write_to_file=&#39;&#39;)</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="co">#       m - m x n matrix of log-transformed intensities, num peptides x num samples</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="co">#       treatment - either a single factor or a data.frame of factors (actual R factors, else code will fail) </span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="co">#                   eg:  bl = gl(3,14,168) # 3 factors, repeat every 14 times, total of 168 samples</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">#                        it = gl(2,7,168)  # 2 factors, repeat every 7 samples, 168 total</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="co">#                        Pi = gl(2,42,168) # 2 factors, repeat twice: 42 of PI+, PI-, PI+, PI-</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="co">#                       grpFactors = data.frame(bl,it,Pi)# factors we would like to preserve in EigenMS</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="co">#       prot.info - 2 column data frame with peptide and protein IDs in that order.</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="co">#                   for metabolites both columns should contain metabolite IDs.</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="co">#       write_to_file -  if a string is passed in, &#39;complete&#39; peptides will be written to that file name</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="co">#                       Some peptides could be eliminated due to too many missing values (if not able to do ANOVA)                      </span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="co"># SUPPLEMENTARY FUNCTION USED BY EIGENMS</span></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="co"># plot top 3 eigentrends with a line at 0.</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="co"># in cse of a single treatment group u and v matrices are switched..</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>plot.eigentrends <span class="ot">=</span> <span class="cf">function</span>(svdr, title1){</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>  v <span class="ot">=</span> svdr<span class="sc">$</span>v</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">=</span> svdr<span class="sc">$</span>d</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>  ss <span class="ot">=</span> d<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>  Tk <span class="ot">=</span> <span class="fu">signif</span>(ss<span class="sc">/</span><span class="fu">sum</span>(ss)<span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>  titles <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Trend &quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="st">&quot; (&quot;</span>, Tk[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>  do.text <span class="ot">=</span> <span class="cf">function</span>(j) <span class="fu">mtext</span>(titles[j], <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">padj=</span><span class="sc">-</span><span class="fl">0.7</span>, <span class="at">adj=</span><span class="dv">1</span>)</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>  range.y <span class="ot">=</span> <span class="fu">range</span>(<span class="fu">as.numeric</span>(v[,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]), <span class="at">na.rm=</span>T)</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>  toplot1_1 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,<span class="dv">1</span>])</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>  toplot1_2 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,<span class="dv">2</span>])</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>  toplot1_3 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,<span class="dv">3</span>])</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_1)), toplot1_1, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">1</span>)</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">title</span>(title1, <span class="at">cex.main =</span> <span class="fl">1.2</span>, <span class="at">font.main=</span> <span class="dv">1</span>, <span class="at">col.main=</span> <span class="st">&quot;purple&quot;</span>, <span class="at">ylab=</span><span class="cn">NULL</span>)</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_2)), toplot1_2, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">2</span>)</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_3)), toplot1_3, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">3</span>)</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(Tk)</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a><span class="co"># plot any 3 eigentrends with a line at 0. Strting at the trend number passed in</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a><span class="co"># pos1 parameter provide the starting index for teh 3 trends</span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="co"># in cse of a single treatment group u and v matrices are switched..</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>plot.eigentrends.start <span class="ot">=</span> <span class="cf">function</span>(svdr, title1, <span class="at">pos1=</span><span class="dv">1</span>){</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>  <span class="co"># No check for valid range of pos1 is performed!!! </span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a>  v <span class="ot">=</span> svdr<span class="sc">$</span>v</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">=</span> svdr<span class="sc">$</span>d</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>  ss <span class="ot">=</span> d<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>  Tk <span class="ot">=</span> <span class="fu">signif</span>(ss<span class="sc">/</span><span class="fu">sum</span>(ss)<span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  pe = signif(d/sum(d, na.rm=T)*100, 2)</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>  titles <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&quot;Trend &quot;</span>, pos1<span class="sc">:</span>(pos1<span class="sc">+</span><span class="dv">3</span>), <span class="st">&quot; (&quot;</span>, Tk[pos1<span class="sc">:</span>(pos1<span class="sc">+</span><span class="dv">3</span>)], <span class="st">&quot;%)&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a>  do.text <span class="ot">=</span> <span class="cf">function</span>(j) <span class="fu">mtext</span>(titles[j], <span class="at">cex=</span><span class="fl">0.7</span>, <span class="at">padj=</span><span class="sc">-</span><span class="fl">0.7</span>, <span class="at">adj=</span><span class="dv">1</span>)</span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>  range.y <span class="ot">=</span> <span class="fu">range</span>(<span class="fu">as.numeric</span>(v[,pos1<span class="sc">:</span>(pos1<span class="sc">+</span><span class="dv">3</span>)]), <span class="at">na.rm=</span>T)</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>  toplot1_1 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,pos1])</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>  toplot1_2 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,(pos1<span class="sc">+</span><span class="dv">1</span>)])</span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a>  toplot1_3 <span class="ot">=</span> <span class="fu">as.numeric</span>(v[,(pos1<span class="sc">+</span><span class="dv">2</span>)])</span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_1)), toplot1_1, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">1</span>)</span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">title</span>(title1, <span class="at">cex.main =</span> <span class="fl">1.2</span>, <span class="at">font.main=</span> <span class="dv">1</span>, <span class="at">col.main=</span> <span class="st">&quot;purple&quot;</span>, <span class="at">ylab=</span><span class="cn">NULL</span>)</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_2)), toplot1_2, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">2</span>)</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(toplot1_3)), toplot1_3, <span class="at">type=</span><span class="st">&#39;b&#39;</span>, <span class="at">ann=</span>F, <span class="at">ylim=</span>range.y)</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">do.text</span>(<span class="dv">3</span>)</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">abline</span>(<span class="at">h=</span><span class="dv">0</span>, <span class="at">lty=</span><span class="dv">3</span>)</span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(Tk)</span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a><span class="co"># not used in EigenMS </span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>make.formula.string <span class="ot">=</span> <span class="cf">function</span>(factors, <span class="at">do.interactions=</span><span class="cn">FALSE</span>){</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>  fs <span class="ot">=</span> <span class="st">&quot;1&quot;</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">length</span>(factors)){</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>    fs <span class="ot">=</span> <span class="fu">paste</span>(factors, <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(do.interactions <span class="sc">&amp;&amp;</span> <span class="fu">length</span>(factors) <span class="sc">&gt;</span> <span class="dv">1</span>)</span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a>      fs <span class="ot">=</span> <span class="fu">paste</span>(<span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">combinations</span>(<span class="fu">length</span>(factors), <span class="dv">2</span>, factors)), <span class="at">stringsAsFactors=</span>F), paste, <span class="at">collapse=</span><span class="st">&quot;*&quot;</span>)), <span class="at">collapse =</span> <span class="st">&quot; + &quot;</span>)</span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(fs)</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="co"># make a string formula to use in &#39;lm&#39; call when computing grp differences to preserve</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a>makeLMFormula <span class="ot">=</span> <span class="cf">function</span>(eff, <span class="at">var_name=</span><span class="st">&#39;&#39;</span>) {</span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a>  <span class="co"># eff - effects used in contrasts</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>  <span class="co"># var_name - for singe factor use var-name that is passed in as variable names, otherwise it has no colnmae</span></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>  <span class="co">#           only used for a single factor</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.factor</span>(eff))</span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>    ndims <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>    cols1 <span class="ot">=</span> var_name <span class="co"># ftemp in EigenMS</span></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>    ndims <span class="ot">=</span> <span class="fu">dim</span>(eff)[<span class="dv">2</span>] </span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>    cols1 <span class="ot">=</span> <span class="fu">colnames</span>(eff)</span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>  lhs <span class="ot">=</span> cols1[<span class="dv">1</span>]</span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a>  lm.fm <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a>  <span class="co"># check if can have a list if only have 1 factor...</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a>  params <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&#39;contrasts=list(&#39;</span>, cols1[<span class="dv">1</span>], <span class="st">&#39;=contr.sum&#39;</span>, <span class="at">sep=</span>)</span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (ndims <span class="sc">&gt;</span> <span class="dv">1</span>) { <span class="co"># removed ndims[2] here, now ndims holds only 1 dimention...</span></span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="fu">length</span>(cols1))</span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a>    {</span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>      lhs <span class="ot">=</span> <span class="fu">paste</span>(lhs, <span class="st">&quot;+&quot;</span>, cols1[ii])  <span class="co"># bl=&quot;contr.sum&quot;,</span></span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a>        params <span class="ot">=</span> <span class="fu">paste</span>(params, <span class="st">&#39;,&#39;</span>, cols1[ii], <span class="st">&#39;=contr.sum&#39;</span>, <span class="at">sep=</span><span class="st">&#39;&#39;</span>)</span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>  params <span class="ot">=</span> <span class="fu">paste</span>(params,<span class="st">&quot;)&quot;</span>) </span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>  lm.formula <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(<span class="st">&#39;~&#39;</span>, lhs))</span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>  lm.fm<span class="sc">$</span>lm.formula <span class="ot">=</span> lm.formula</span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>  lm.fm<span class="sc">$</span>lm.params <span class="ot">=</span> params</span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(lm.fm)</span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>}   </span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a><span class="co"># First portion of EigenMS: identify bias trends </span></span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a>eig_norm1 <span class="ot">=</span> <span class="cf">function</span>(m, treatment, prot.info, <span class="at">write_to_file=</span><span class="st">&#39;&#39;</span>){</span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify significant eigentrends, allow the user to adjust the number (with causion! if desired)</span></span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a><span class="co"># before normalizing with eig_norm2</span></span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a><span class="co"># Input:</span></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a><span class="co">#   m: An m x n (peptides x samples) matrix of expression data, log-transformed!</span></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a><span class="co">#      peptide and protein identifiers come from the get.ProtInfo()</span></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a><span class="co">#   treatment:  either a single factor indicating the treatment group of each sample i.e. [1 1 1 1 2 2 2 2...]</span></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a><span class="co">#               or a frame of factors:  treatment= data.frame(cbind(data.frame(Group), data.frame(Time)) </span></span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a><span class="co">#   prot.info: 2+ colum data frame, pepID, prID columns IN THAT ORDER. </span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a><span class="co">#              IMPORTANT: pepIDs must be unique identifiers and will be used as Row Names </span></span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a><span class="co">#              If normalizing non-proteomics data, create a column such as: paste(&#39;ID_&#39;,seq(1:num_rows), sep=&#39;&#39;)</span></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a><span class="co">#              Same can be dome for ProtIDs, these are not used for normalization but are kept for future analyses </span></span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a><span class="co">#   write_to_file=&#39;&#39; - if a string is passed in, &#39;complete&#39; peptides (peptides with NO missing observations)</span></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a><span class="co">#              will be written to that file name</span></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a><span class="co">#                    </span></span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a><span class="co"># Output: list of:</span></span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a><span class="co">#   m, treatment, prot.info, grp - initial parameters returned for futre reference </span></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="co">#   my.svd - matrices produced by SVD </span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a><span class="co">#   pres - matrix of peptides that can be normalized, i.e. have enough observations for ANOVA, </span></span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a><span class="co">#   n.treatment - number of factors passed in</span></span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a><span class="co">#   n.u.treatment - number of unique treatment facotr combinations, eg: </span></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a><span class="co">#                   Factor A: a a a a c c c c</span></span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a><span class="co">#                   Factor B: 1 1 2 2 1 1 2 2</span></span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a><span class="co">#                   then:  n.treatment = 2; n.u.treatment = 4</span></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a><span class="co">#   h.c - bias trends </span></span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a><span class="co">#   present - names/IDs of peptides on pres</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a><span class="co">#   complete - complete peptides, no missing values, these were used to compute SVD</span></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a><span class="co">#   toplot1 - trends automatically produced, if one wanted to plot at later time. </span></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a><span class="co">#   Tk - scores for each bias trend </span></span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a><span class="co">#   ncompl - number of complete peptides with no missing observations</span></span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;Data dimentions: &quot;</span>)  </span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(m))</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>  <span class="co"># check if treatment is a &#39;factor&#39; vs data.frame&#39;, i.e. single vs multiple factors</span></span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">class</span>(treatment) <span class="sc">==</span> <span class="st">&quot;factor&quot;</span>) { <span class="co"># TRUE if one factor</span></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a>     n.treatment <span class="ot">=</span> <span class="dv">1</span> <span class="co"># length(treatment)</span></span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a>     n.u.treatment <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(treatment))[<span class="dv">1</span>]</span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> { <span class="co"># data.frame</span></span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>    n.treatment <span class="ot">=</span> <span class="fu">dim</span>(treatment)[<span class="dv">2</span>]</span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a>    n.u.treatment <span class="ot">=</span> <span class="fu">dim</span>(<span class="fu">unique</span>(treatment))[<span class="dv">1</span>] <span class="co"># all possible tretment combinations</span></span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a>  <span class="co"># convert m to a matrix from data.frame</span></span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">=</span> <span class="fu">as.matrix</span>(m) <span class="co"># no loss of information</span></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a>  <span class="co"># filter out min.missing, here just counting missing values</span></span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if 1+ treatment completely missing, cannot do ANOVA, thus cannot preserve grp diff.</span></span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a>  <span class="co"># IMPORTANT: we create a composite grp = number of unique combinations of all groups, only for </span></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a>  <span class="co"># &#39;nested&#39; groups for single layer group is left as it is </span></span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a>  grpFactors <span class="ot">=</span> treatment <span class="co"># temporary var, leftover from old times...</span></span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a>  nGrpFactors <span class="ot">=</span> n.treatment <span class="co"># length(colnames(treatment)) # not good: dim(grpFactors)</span></span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(nGrpFactors <span class="sc">&gt;</span> <span class="dv">1</span>) { <span class="co"># got nested factors</span></span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>    ugrps <span class="ot">=</span> <span class="fu">unique</span>(grpFactors)</span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a>    udims <span class="ot">=</span> <span class="fu">dim</span>(ugrps)</span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>    grp <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>udims[<span class="dv">1</span>]) {</span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>      pos <span class="ot">=</span> grpFactors[,<span class="dv">1</span>] <span class="sc">==</span> ugrps[ii,<span class="dv">1</span>] <span class="co"># set to initial value</span></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>      <span class="cf">for</span>(jj <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>udims[<span class="dv">2</span>]) { </span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>        pos <span class="ot">=</span> pos <span class="sc">&amp;</span> grpFactors[,jj] <span class="sc">==</span> ugrps[ii,jj]</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>      grp[pos] <span class="ot">=</span> <span class="fu">rep</span>(ii, <span class="fu">sum</span>(pos))</span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>    grp <span class="ot">=</span> <span class="fu">as.factor</span>(grp)</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a>    grp <span class="ot">=</span> treatment</span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a>  nobs <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="fu">c</span>(<span class="fu">nrow</span>(m), <span class="fu">length</span>(<span class="fu">unique</span>(grp)))) <span class="co"># noobs = number of observations </span></span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&#39;Treatmenet groups:&#39;</span>)</span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(grp)</span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(m)) {</span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span>(jj <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(<span class="fu">unique</span>(grp))) {</span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a>      nobs[ii,jj] <span class="ot">=</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(m[ii, grp<span class="sc">==</span><span class="fu">unique</span>(grp)[jj]])) <span class="co"># total number of groups num(g1) * num(g2) * ...</span></span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a>    } </span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a>  } </span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a>  <span class="co"># now &#39;remove&#39; peptides with missing groups</span></span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a>  present.min <span class="ot">=</span> <span class="fu">apply</span>(nobs, <span class="dv">1</span>, min) <span class="co"># number present in each group</span></span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a>  ii <span class="ot">=</span> present.min <span class="sc">==</span> <span class="dv">0</span>   <span class="co"># 1+ obs present in ALL of the groups</span></span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a>  nmiss <span class="ot">=</span> <span class="fu">sum</span>(present.min <span class="sc">==</span> <span class="dv">0</span>) <span class="co"># not used, one value of how many peptides have 1+ grp missing completely</span></span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>  pmiss <span class="ot">=</span> <span class="fu">rbind</span>(m[ii,]) <span class="co"># these have 1+ grp missing !!!!</span></span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a>  <span class="co"># rownames must be UNIQUE, if have possible duplicates: use &#39;ii&#39; ?</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(pmiss) <span class="ot">=</span> prot.info[ii,<span class="dv">1</span>]  <span class="co"># set rownames, </span></span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create matrix for peptides with enough observations for ANOVA</span></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a>  <span class="co"># &#39;present&#39; are names of the peptides (pepID) and &#39;pres&#39; are abundances</span></span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span><span class="al">NOTE</span><span class="co">: ! negates the proteins, so we get ones that have 1+ obs in each group </span></span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a>  present <span class="ot">=</span> prot.info[<span class="fu">which</span>(<span class="sc">!</span>prot.info[,<span class="dv">1</span>] <span class="sc">%in%</span> <span class="fu">rownames</span>(pmiss)), ] <span class="co"># rownames OK</span></span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pres = m[which(!rownames(m) %in% rownames(pmiss)), ]</span></span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a>  pres <span class="ot">=</span> m[<span class="fu">which</span>(<span class="sc">!</span>prot.info[,<span class="dv">1</span>] <span class="sc">%in%</span> <span class="fu">rownames</span>(pmiss)), ] <span class="co"># is this OK?</span></span>
<span id="cb1-263"><a href="#cb1-263" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(pres) <span class="ot">=</span> prot.info[<span class="fu">which</span>(<span class="sc">!</span>prot.info[,<span class="dv">1</span>] <span class="sc">%in%</span> <span class="fu">rownames</span>(pmiss)),<span class="dv">1</span>]</span>
<span id="cb1-264"><a href="#cb1-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-265"><a href="#cb1-265" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&#39;Selecting complete peptides&#39;</span>)</span>
<span id="cb1-266"><a href="#cb1-266" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Should issue an error message if we have NO complete peptides.</span></span>
<span id="cb1-267"><a href="#cb1-267" aria-hidden="true" tabindex="-1"></a>  <span class="co"># select only &#39;complete&#39; peptides, no missing values</span></span>
<span id="cb1-268"><a href="#cb1-268" aria-hidden="true" tabindex="-1"></a>  nobs <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="fu">nrow</span>(pres)) <span class="co"># reassign noobs to dims of &#39;present&#39; </span></span>
<span id="cb1-269"><a href="#cb1-269" aria-hidden="true" tabindex="-1"></a>  numiter <span class="ot">=</span> <span class="fu">nrow</span>(pres)</span>
<span id="cb1-270"><a href="#cb1-270" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>numiter) {</span>
<span id="cb1-271"><a href="#cb1-271" aria-hidden="true" tabindex="-1"></a>    <span class="co"># if(ii %% 100 == 0) { print(ii) }</span></span>
<span id="cb1-272"><a href="#cb1-272" aria-hidden="true" tabindex="-1"></a>    nobs[ii] <span class="ot">=</span> <span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(pres[ii,]))</span>
<span id="cb1-273"><a href="#cb1-273" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-274"><a href="#cb1-274" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-275"><a href="#cb1-275" aria-hidden="true" tabindex="-1"></a>  iii <span class="ot">=</span> nobs <span class="sc">==</span> <span class="fu">ncol</span>(pres)</span>
<span id="cb1-276"><a href="#cb1-276" aria-hidden="true" tabindex="-1"></a>  complete <span class="ot">=</span> <span class="fu">rbind</span>(pres[iii,])</span>
<span id="cb1-277"><a href="#cb1-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-278"><a href="#cb1-278" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  write out a file of complete peptides if file name is passed in</span></span>
<span id="cb1-279"><a href="#cb1-279" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(write_to_file <span class="sc">!=</span> <span class="st">&#39;&#39;</span>) {</span>
<span id="cb1-280"><a href="#cb1-280" aria-hidden="true" tabindex="-1"></a>    <span class="fu">write.table</span>(complete, <span class="at">file =</span> write_to_file, <span class="at">append =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1-281"><a href="#cb1-281" aria-hidden="true" tabindex="-1"></a>             <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,</span>
<span id="cb1-282"><a href="#cb1-282" aria-hidden="true" tabindex="-1"></a>             <span class="at">eol =</span> <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>, <span class="at">na =</span> <span class="st">&quot;NaN&quot;</span>, <span class="at">dec =</span> <span class="st">&quot;.&quot;</span>, <span class="at">row.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb1-283"><a href="#cb1-283" aria-hidden="true" tabindex="-1"></a>             <span class="at">col.names =</span> <span class="cn">TRUE</span>, <span class="at">qmethod =</span> <span class="fu">c</span>(<span class="st">&quot;escape&quot;</span>, <span class="st">&quot;double&quot;</span>))</span>
<span id="cb1-284"><a href="#cb1-284" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-285"><a href="#cb1-285" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-286"><a href="#cb1-286" aria-hidden="true" tabindex="-1"></a>  <span class="co"># compute bias with &#39;complete&#39; matrix and residuals from &#39;present&#39; </span></span>
<span id="cb1-287"><a href="#cb1-287" aria-hidden="true" tabindex="-1"></a>  <span class="co"># calculate eigenpeptides for &#39;complete&#39; data only</span></span>
<span id="cb1-288"><a href="#cb1-288" aria-hidden="true" tabindex="-1"></a>  <span class="co"># if have only 1 group, we do not need to preserve group differernces, everything is the same group, ex: QC samples</span></span>
<span id="cb1-289"><a href="#cb1-289" aria-hidden="true" tabindex="-1"></a>  <span class="co"># contrasts will fail if have only 1 group, thus have else</span></span>
<span id="cb1-290"><a href="#cb1-290" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(n.u.treatment <span class="sc">&gt;</span> <span class="dv">1</span>) { </span>
<span id="cb1-291"><a href="#cb1-291" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Got 2+ treatment grps&#39;</span>)</span>
<span id="cb1-292"><a href="#cb1-292" aria-hidden="true" tabindex="-1"></a>    <span class="co"># check to see if we have multiple factors</span></span>
<span id="cb1-293"><a href="#cb1-293" aria-hidden="true" tabindex="-1"></a>      grpdim <span class="ot">=</span> <span class="fu">dim</span>(treatment)</span>
<span id="cb1-294"><a href="#cb1-294" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-295"><a href="#cb1-295" aria-hidden="true" tabindex="-1"></a>      lm.fm <span class="ot">=</span> <span class="fu">makeLMFormula</span>(treatment, <span class="st">&#39;TREAT&#39;</span>) <span class="co"># using general function that can accomodate for 1+ number of factors</span></span>
<span id="cb1-296"><a href="#cb1-296" aria-hidden="true" tabindex="-1"></a>    TREAT <span class="ot">=</span> treatment</span>
<span id="cb1-297"><a href="#cb1-297" aria-hidden="true" tabindex="-1"></a>    TREAT <span class="ot">=</span> <span class="fu">data.frame</span>(treatment) <span class="co"># temp var to work if we got only 1 treatment vector.</span></span>
<span id="cb1-298"><a href="#cb1-298" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">class</span>(treatment) <span class="sc">==</span> <span class="st">&quot;factor&quot;</span>) {</span>
<span id="cb1-299"><a href="#cb1-299" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(TREAT) <span class="ot">=</span> <span class="st">&quot;TREAT&quot;</span></span>
<span id="cb1-300"><a href="#cb1-300" aria-hidden="true" tabindex="-1"></a>     } <span class="cf">else</span> {</span>
<span id="cb1-301"><a href="#cb1-301" aria-hidden="true" tabindex="-1"></a>      <span class="fu">colnames</span>(TREAT) <span class="ot">=</span> <span class="fu">colnames</span>(treatment)</span>
<span id="cb1-302"><a href="#cb1-302" aria-hidden="true" tabindex="-1"></a>    }     </span>
<span id="cb1-303"><a href="#cb1-303" aria-hidden="true" tabindex="-1"></a>    <span class="fu">attach</span>(TREAT)</span>
<span id="cb1-304"><a href="#cb1-304" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-305"><a href="#cb1-305" aria-hidden="true" tabindex="-1"></a>    mod.c <span class="ot">=</span> <span class="fu">model.matrix</span>(lm.fm<span class="sc">$</span>lm.formula, <span class="at">data=</span>TREAT, <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>lm.fm<span class="sc">$</span>lm.params))) </span>
<span id="cb1-306"><a href="#cb1-306" aria-hidden="true" tabindex="-1"></a>      Y.c <span class="ot">=</span> <span class="fu">as.matrix</span>(complete)</span>
<span id="cb1-307"><a href="#cb1-307" aria-hidden="true" tabindex="-1"></a>      <span class="fu">options</span>(<span class="at">warn =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb1-308"><a href="#cb1-308" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb1-309"><a href="#cb1-309" aria-hidden="true" tabindex="-1"></a>    <span class="co"># use lm() to get residuals</span></span>
<span id="cb1-310"><a href="#cb1-310" aria-hidden="true" tabindex="-1"></a>    formula1 <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&#39;t(Y.c)~&#39;</span>, <span class="fu">as.character</span>(lm.fm<span class="sc">$</span>lm.formula)[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb1-311"><a href="#cb1-311" aria-hidden="true" tabindex="-1"></a>    TREAT <span class="ot">=</span> treatment</span>
<span id="cb1-312"><a href="#cb1-312" aria-hidden="true" tabindex="-1"></a>    fit_lmAll <span class="ot">=</span> <span class="fu">lm</span>(<span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>formula1)))</span>
<span id="cb1-313"><a href="#cb1-313" aria-hidden="true" tabindex="-1"></a>    R.c <span class="ot">=</span> <span class="fu">residuals</span>(fit_lmAll)  <span class="co"># Oct 2 messing with residuals...</span></span>
<span id="cb1-314"><a href="#cb1-314" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {  <span class="co"># 1 group only, set residuals to original matrix</span></span>
<span id="cb1-315"><a href="#cb1-315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(<span class="st">&#39;Got 1 treatment grp&#39;</span>)</span>
<span id="cb1-316"><a href="#cb1-316" aria-hidden="true" tabindex="-1"></a>      mod.c <span class="ot">=</span> <span class="fu">as.numeric</span>(<span class="fu">t</span>(treatment))</span>
<span id="cb1-317"><a href="#cb1-317" aria-hidden="true" tabindex="-1"></a>    R.c <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">as.matrix</span>(complete))  <span class="co"># needs to be transposed to match the matrix returned from lm</span></span>
<span id="cb1-318"><a href="#cb1-318" aria-hidden="true" tabindex="-1"></a>    TREAT <span class="ot">=</span> treatment</span>
<span id="cb1-319"><a href="#cb1-319" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-320"><a href="#cb1-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-321"><a href="#cb1-321" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&#39;Computing SVD, estimating Eigentrends...&#39;</span>) <span class="co"># let user know what is going on</span></span>
<span id="cb1-322"><a href="#cb1-322" aria-hidden="true" tabindex="-1"></a>  <span class="co"># residuals are centered around 0, here center samples not peptides/metabolites</span></span>
<span id="cb1-323"><a href="#cb1-323" aria-hidden="true" tabindex="-1"></a>  <span class="co"># centering is basic normalization</span></span>
<span id="cb1-324"><a href="#cb1-324" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-325"><a href="#cb1-325" aria-hidden="true" tabindex="-1"></a>  R.c_center <span class="ot">=</span> <span class="fu">scale</span>(R.c, <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>)  <span class="co"># t(scale(t(R.c), center = TRUE, scale = FALSE))</span></span>
<span id="cb1-326"><a href="#cb1-326" aria-hidden="true" tabindex="-1"></a>  my.svd <span class="ot">=</span> <span class="fu">svd</span>(R.c_center)  <span class="co"># can use wrapper below to chek if SVD has a problem...</span></span>
<span id="cb1-327"><a href="#cb1-327" aria-hidden="true" tabindex="-1"></a>  temp <span class="ot">=</span> my.svd<span class="sc">$</span>u</span>
<span id="cb1-328"><a href="#cb1-328" aria-hidden="true" tabindex="-1"></a>  my.svd<span class="sc">$</span>u <span class="ot">=</span> my.svd<span class="sc">$</span>v</span>
<span id="cb1-329"><a href="#cb1-329" aria-hidden="true" tabindex="-1"></a>  my.svd<span class="sc">$</span>v <span class="ot">=</span> temp</span>
<span id="cb1-330"><a href="#cb1-330" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-331"><a href="#cb1-331" aria-hidden="true" tabindex="-1"></a>  <span class="co">#identify number of eigenvalues that account for a significant amount of residual variation</span></span>
<span id="cb1-332"><a href="#cb1-332" aria-hidden="true" tabindex="-1"></a>  numcompletepep <span class="ot">=</span> <span class="fu">dim</span>(complete)[<span class="dv">1</span>] <span class="co"># save to return to the user as part of the return list  </span></span>
<span id="cb1-333"><a href="#cb1-333" aria-hidden="true" tabindex="-1"></a>  <span class="co"># this is important info for publications</span></span>
<span id="cb1-334"><a href="#cb1-334" aria-hidden="true" tabindex="-1"></a>  <span class="co"># tell users how many peptides/metabolites the trends are based on</span></span>
<span id="cb1-335"><a href="#cb1-335" aria-hidden="true" tabindex="-1"></a>  <span class="co"># can also be determined by doing dim(return_value_fromEIg_norm1$pres)</span></span>
<span id="cb1-336"><a href="#cb1-336" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-337"><a href="#cb1-337" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Number of treatments: &#39;</span>, n.u.treatment))</span>
<span id="cb1-338"><a href="#cb1-338" aria-hidden="true" tabindex="-1"></a>  h.c <span class="ot">=</span> <span class="fu">sva.id</span>(complete, treatment, n.u.treatment, <span class="at">lm.fm=</span>lm.fm,<span class="at">seed=</span><span class="dv">1234</span>)<span class="sc">$</span>n.sv</span>
<span id="cb1-339"><a href="#cb1-339" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Number of significant eigenpeptides/trends&quot;</span>, h.c) )</span>
<span id="cb1-340"><a href="#cb1-340" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-341"><a href="#cb1-341" aria-hidden="true" tabindex="-1"></a>  <span class="co"># show RAW trends</span></span>
<span id="cb1-342"><a href="#cb1-342" aria-hidden="true" tabindex="-1"></a>  <span class="co"># center each peptide around zero (subtract its mean across samples)</span></span>
<span id="cb1-343"><a href="#cb1-343" aria-hidden="true" tabindex="-1"></a>  complete_center <span class="ot">=</span> <span class="fu">scale</span>(<span class="fu">t</span>(complete), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>)</span>
<span id="cb1-344"><a href="#cb1-344" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&#39;Preparing to plot...&#39;</span>)</span>
<span id="cb1-345"><a href="#cb1-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-346"><a href="#cb1-346" aria-hidden="true" tabindex="-1"></a>  n.u.treatment</span>
<span id="cb1-347"><a href="#cb1-347" aria-hidden="true" tabindex="-1"></a>  toplot1 <span class="ot">=</span> <span class="fu">svd</span>(complete_center) <span class="co"># scales above</span></span>
<span id="cb1-348"><a href="#cb1-348" aria-hidden="true" tabindex="-1"></a>  temp <span class="ot">=</span> toplot1<span class="sc">$</span>u</span>
<span id="cb1-349"><a href="#cb1-349" aria-hidden="true" tabindex="-1"></a>  toplot1<span class="sc">$</span>u <span class="ot">=</span> toplot1<span class="sc">$</span>v</span>
<span id="cb1-350"><a href="#cb1-350" aria-hidden="true" tabindex="-1"></a>  toplot1<span class="sc">$</span>v <span class="ot">=</span> temp</span>
<span id="cb1-351"><a href="#cb1-351" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-352"><a href="#cb1-352" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfcol=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb1-353"><a href="#cb1-353" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb1-354"><a href="#cb1-354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.eigentrends</span>(toplot1, <span class="st">&quot;Raw Data&quot;</span>)</span>
<span id="cb1-355"><a href="#cb1-355" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.eigentrends</span>(my.svd, <span class="st">&quot;Residual Data&quot;</span>)</span>
<span id="cb1-356"><a href="#cb1-356" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-357"><a href="#cb1-357" aria-hidden="true" tabindex="-1"></a>  d <span class="ot">=</span> my.svd<span class="sc">$</span>d;  ss <span class="ot">=</span> d<span class="sc">^</span><span class="dv">2</span>;</span>
<span id="cb1-358"><a href="#cb1-358" aria-hidden="true" tabindex="-1"></a>  Tk <span class="ot">=</span> <span class="fu">signif</span>(ss<span class="sc">/</span><span class="fu">sum</span>(ss)<span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span>
<span id="cb1-359"><a href="#cb1-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-360"><a href="#cb1-360" aria-hidden="true" tabindex="-1"></a>  retval <span class="ot">=</span> <span class="fu">list</span>(<span class="at">m=</span>m, <span class="at">treatment=</span>treatment, <span class="at">my.svd=</span>my.svd,</span>
<span id="cb1-361"><a href="#cb1-361" aria-hidden="true" tabindex="-1"></a>            <span class="at">pres=</span>pres, <span class="at">n.treatment=</span>n.treatment, <span class="at">n.u.treatment=</span>n.u.treatment,</span>
<span id="cb1-362"><a href="#cb1-362" aria-hidden="true" tabindex="-1"></a>            <span class="at">h.c=</span>h.c, <span class="at">present=</span>present, <span class="at">prot.info=</span>prot.info,</span>
<span id="cb1-363"><a href="#cb1-363" aria-hidden="true" tabindex="-1"></a>            <span class="at">complete=</span>complete, <span class="at">toplot1=</span>toplot1, <span class="at">Tk=</span>Tk, <span class="at">ncompl=</span>numcompletepep,</span>
<span id="cb1-364"><a href="#cb1-364" aria-hidden="true" tabindex="-1"></a>            <span class="at">grp=</span>grp) </span>
<span id="cb1-365"><a href="#cb1-365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(retval)</span>
<span id="cb1-366"><a href="#cb1-366" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-367"><a href="#cb1-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-368"><a href="#cb1-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-369"><a href="#cb1-369" aria-hidden="true" tabindex="-1"></a><span class="co"># Second portion of EigenMS: remove effects of bias </span></span>
<span id="cb1-370"><a href="#cb1-370" aria-hidden="true" tabindex="-1"></a><span class="co"># split into 2 functions to allo the user to examine the bias trends and be able to change the number</span></span>
<span id="cb1-371"><a href="#cb1-371" aria-hidden="true" tabindex="-1"></a><span class="co"># by resetting h.c returned by eig_norm1(return_value_from_eig_norm1)</span></span>
<span id="cb1-372"><a href="#cb1-372" aria-hidden="true" tabindex="-1"></a>eig_norm2 <span class="ot">=</span> <span class="cf">function</span>(rv) { </span>
<span id="cb1-373"><a href="#cb1-373" aria-hidden="true" tabindex="-1"></a>  <span class="co"># UNPUT:</span></span>
<span id="cb1-374"><a href="#cb1-374" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   rv - return value from the eig_norm1</span></span>
<span id="cb1-375"><a href="#cb1-375" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   if user wants to change the number of bias trends that will be eliminated h.c in rv should </span></span>
<span id="cb1-376"><a href="#cb1-376" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   be updates to the desired number</span></span>
<span id="cb1-377"><a href="#cb1-377" aria-hidden="true" tabindex="-1"></a>  <span class="co"># </span></span>
<span id="cb1-378"><a href="#cb1-378" aria-hidden="true" tabindex="-1"></a>  <span class="co"># OUTPUT: </span></span>
<span id="cb1-379"><a href="#cb1-379" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   normalized - matrix of normalized abundances with 2 columns of protein and peptdie names</span></span>
<span id="cb1-380"><a href="#cb1-380" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   norm_m - matrix of normalized abundances, no extra columns  </span></span>
<span id="cb1-381"><a href="#cb1-381" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   eigentrends - found in raw data, bias trendsup to h.c</span></span>
<span id="cb1-382"><a href="#cb1-382" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   rescrange - rescaling range for the addition of the while noise to avoid overfitting </span></span>
<span id="cb1-383"><a href="#cb1-383" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   norm.svd - trends in normalized data, if one wanted to plot at later time. </span></span>
<span id="cb1-384"><a href="#cb1-384" aria-hidden="true" tabindex="-1"></a>  <span class="co">#   exPeps - excluded peptides - excluded due to exception in fitting a linear model</span></span>
<span id="cb1-385"><a href="#cb1-385" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-386"><a href="#cb1-386" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">=</span> rv<span class="sc">$</span>pres <span class="co"># yuliya: use pres matrix, as we cannot deal with m anyways, need to narrow it down to &#39;complete&#39; peptides</span></span>
<span id="cb1-387"><a href="#cb1-387" aria-hidden="true" tabindex="-1"></a>  treatment <span class="ot">=</span> rv<span class="sc">$</span>treatment</span>
<span id="cb1-388"><a href="#cb1-388" aria-hidden="true" tabindex="-1"></a>  my.svd <span class="ot">=</span> rv<span class="sc">$</span>my.svd</span>
<span id="cb1-389"><a href="#cb1-389" aria-hidden="true" tabindex="-1"></a>  pres <span class="ot">=</span> rv<span class="sc">$</span>pres</span>
<span id="cb1-390"><a href="#cb1-390" aria-hidden="true" tabindex="-1"></a>  n.treatment <span class="ot">=</span> rv<span class="sc">$</span>n.treatment</span>
<span id="cb1-391"><a href="#cb1-391" aria-hidden="true" tabindex="-1"></a>  n.u.treatment <span class="ot">=</span> rv<span class="sc">$</span>n.u.treatment </span>
<span id="cb1-392"><a href="#cb1-392" aria-hidden="true" tabindex="-1"></a>  numFact <span class="ot">=</span> <span class="fu">dim</span>(rv<span class="sc">$</span>treatment)[<span class="dv">2</span>]</span>
<span id="cb1-393"><a href="#cb1-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Unique number of treatment combinations:&#39;</span>, n.u.treatment) )</span>
<span id="cb1-394"><a href="#cb1-394" aria-hidden="true" tabindex="-1"></a>  h.c <span class="ot">=</span> rv<span class="sc">$</span>h.c</span>
<span id="cb1-395"><a href="#cb1-395" aria-hidden="true" tabindex="-1"></a>  present <span class="ot">=</span> rv<span class="sc">$</span>present</span>
<span id="cb1-396"><a href="#cb1-396" aria-hidden="true" tabindex="-1"></a>  toplot1 <span class="ot">=</span> rv<span class="sc">$</span>toplot1</span>
<span id="cb1-397"><a href="#cb1-397" aria-hidden="true" tabindex="-1"></a>  <span class="co"># vector of indicators of peptides that threw exeptions </span></span>
<span id="cb1-398"><a href="#cb1-398" aria-hidden="true" tabindex="-1"></a>  exPeps <span class="ot">=</span> <span class="fu">vector</span>(<span class="at">mode =</span> <span class="st">&quot;numeric&quot;</span>, <span class="at">length =</span> <span class="fu">nrow</span>(pres))</span>
<span id="cb1-399"><a href="#cb1-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-400"><a href="#cb1-400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;Normalizing...&quot;</span>)</span>
<span id="cb1-401"><a href="#cb1-401" aria-hidden="true" tabindex="-1"></a>  treatment <span class="ot">=</span> <span class="fu">data.frame</span>(treatment) <span class="co"># does this need to be done?</span></span>
<span id="cb1-402"><a href="#cb1-402" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(n.u.treatment <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb1-403"><a href="#cb1-403" aria-hidden="true" tabindex="-1"></a>    lm.fm <span class="ot">=</span> <span class="fu">makeLMFormula</span>(treatment, <span class="st">&#39;ftemp&#39;</span>)</span>
<span id="cb1-404"><a href="#cb1-404" aria-hidden="true" tabindex="-1"></a>    mtmp <span class="ot">=</span> <span class="fu">model.matrix</span>(lm.fm<span class="sc">$</span>lm.formula, <span class="at">data=</span>treatment, <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>lm.fm<span class="sc">$</span>lm.params)))  <span class="co">#contrasts=list(bl=&quot;contr.sum&quot;, it=&quot;contr.sum&quot;,Pi=&quot;contr.sum&quot;, tp=&quot;contr.sum&quot;))</span></span>
<span id="cb1-405"><a href="#cb1-405" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {  <span class="co"># have 1 treatment group</span></span>
<span id="cb1-406"><a href="#cb1-406" aria-hidden="true" tabindex="-1"></a>    mtmp <span class="ot">=</span> treatment <span class="co"># as.numeric(t(treatment)) </span></span>
<span id="cb1-407"><a href="#cb1-407" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-408"><a href="#cb1-408" aria-hidden="true" tabindex="-1"></a>  <span class="co"># above needed to know how many values will get back for some matrices</span></span>
<span id="cb1-409"><a href="#cb1-409" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create some variables:</span></span>
<span id="cb1-410"><a href="#cb1-410" aria-hidden="true" tabindex="-1"></a>  betahat <span class="ot">=</span> <span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span><span class="fu">dim</span>(mtmp)[<span class="dv">2</span>],<span class="at">ncol=</span><span class="fu">nrow</span>(pres)) </span>
<span id="cb1-411"><a href="#cb1-411" aria-hidden="true" tabindex="-1"></a>  newR <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="fu">c</span>(<span class="fu">nrow</span>(pres), <span class="fu">ncol</span>(pres))) <span class="co">#, n.treatment))</span></span>
<span id="cb1-412"><a href="#cb1-412" aria-hidden="true" tabindex="-1"></a>  norm_m <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="fu">c</span>(<span class="fu">nrow</span>(pres), <span class="fu">ncol</span>(pres))) <span class="co"># , n.treatment))</span></span>
<span id="cb1-413"><a href="#cb1-413" aria-hidden="true" tabindex="-1"></a>  numsamp <span class="ot">=</span> <span class="fu">dim</span>(pres)[<span class="dv">2</span>]</span>
<span id="cb1-414"><a href="#cb1-414" aria-hidden="true" tabindex="-1"></a>  numpep <span class="ot">=</span> <span class="fu">dim</span>(pres)[<span class="dv">1</span>]</span>
<span id="cb1-415"><a href="#cb1-415" aria-hidden="true" tabindex="-1"></a>  betahat_n <span class="ot">=</span> <span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="at">nrow=</span><span class="fu">dim</span>(mtmp)[<span class="dv">2</span>],<span class="at">ncol=</span><span class="fu">nrow</span>(pres))</span>
<span id="cb1-416"><a href="#cb1-416" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rm</span>(mtmp) </span>
<span id="cb1-417"><a href="#cb1-417" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-418"><a href="#cb1-418" aria-hidden="true" tabindex="-1"></a>  V0 <span class="ot">=</span> my.svd<span class="sc">$</span>v[,<span class="dv">1</span><span class="sc">:</span>h.c,drop<span class="ot">=</span>F]   <span class="co"># residual eigenpeptides</span></span>
<span id="cb1-419"><a href="#cb1-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-420"><a href="#cb1-420" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(n.u.treatment <span class="sc">==</span> <span class="dv">1</span>) { <span class="co"># got 1 treatment group</span></span>
<span id="cb1-421"><a href="#cb1-421" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pres)) {</span>
<span id="cb1-422"><a href="#cb1-422" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(ii<span class="sc">%%</span><span class="dv">250</span> <span class="sc">==</span> <span class="dv">0</span>) { <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Processing peptide &#39;</span>,ii))  }</span>
<span id="cb1-423"><a href="#cb1-423" aria-hidden="true" tabindex="-1"></a>      pep <span class="ot">=</span> pres[ii, ] </span>
<span id="cb1-424"><a href="#cb1-424" aria-hidden="true" tabindex="-1"></a>      pos <span class="ot">=</span> <span class="sc">!</span><span class="fu">is.na</span>(pep)</span>
<span id="cb1-425"><a href="#cb1-425" aria-hidden="true" tabindex="-1"></a>      peptemp <span class="ot">=</span> <span class="fu">as.matrix</span>(pep[pos]) <span class="co"># take only the observed values</span></span>
<span id="cb1-426"><a href="#cb1-426" aria-hidden="true" tabindex="-1"></a>      resm <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, numsamp) </span>
<span id="cb1-427"><a href="#cb1-427" aria-hidden="true" tabindex="-1"></a>      resm[pos] <span class="ot">=</span> <span class="fu">as.numeric</span>(pep[pos])</span>
<span id="cb1-428"><a href="#cb1-428" aria-hidden="true" tabindex="-1"></a>      bias <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, numsamp)</span>
<span id="cb1-429"><a href="#cb1-429" aria-hidden="true" tabindex="-1"></a>      bias[pos] <span class="ot">=</span> resm[pos] <span class="sc">%*%</span> V0[pos,] <span class="sc">%*%</span> <span class="fu">t</span>(V0[pos,])</span>
<span id="cb1-430"><a href="#cb1-430" aria-hidden="true" tabindex="-1"></a>      norm_m[ii, ] <span class="ot">=</span> <span class="fu">as.numeric</span>(pep <span class="sc">-</span> bias)</span>
<span id="cb1-431"><a href="#cb1-431" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-432"><a href="#cb1-432" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-433"><a href="#cb1-433" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> { <span class="co"># got 2+ treatment groups</span></span>
<span id="cb1-434"><a href="#cb1-434" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pres)) {</span>
<span id="cb1-435"><a href="#cb1-435" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(ii <span class="sc">%%</span> <span class="dv">100</span> <span class="sc">==</span> <span class="dv">0</span>) { <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Processing peptide &#39;</span>,ii))  }</span>
<span id="cb1-436"><a href="#cb1-436" aria-hidden="true" tabindex="-1"></a>      pep <span class="ot">=</span> pres[ii, ] </span>
<span id="cb1-437"><a href="#cb1-437" aria-hidden="true" tabindex="-1"></a>      pos <span class="ot">=</span> <span class="sc">!</span><span class="fu">is.na</span>(pep)</span>
<span id="cb1-438"><a href="#cb1-438" aria-hidden="true" tabindex="-1"></a>      peptemp <span class="ot">=</span> <span class="fu">as.matrix</span>(pep[pos]) <span class="co"># take only the observed values, may not be needed in R? but this works</span></span>
<span id="cb1-439"><a href="#cb1-439" aria-hidden="true" tabindex="-1"></a>      ftemp <span class="ot">=</span> treatment[pos,]</span>
<span id="cb1-440"><a href="#cb1-440" aria-hidden="true" tabindex="-1"></a>      ftemp <span class="ot">=</span> <span class="fu">data.frame</span>(ftemp)</span>
<span id="cb1-441"><a href="#cb1-441" aria-hidden="true" tabindex="-1"></a>      <span class="do">#### use try, not entirely sure if need for modt, need it for solve lm?!</span></span>
<span id="cb1-442"><a href="#cb1-442" aria-hidden="true" tabindex="-1"></a>      <span class="fu">options</span>(<span class="at">warn =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb1-443"><a href="#cb1-443" aria-hidden="true" tabindex="-1"></a>      lm.fm <span class="ot">=</span> <span class="fu">makeLMFormula</span>(ftemp, <span class="st">&#39;ftemp&#39;</span>) <span class="co"># using general function that can accomodate for 1+ number of factors</span></span>
<span id="cb1-444"><a href="#cb1-444" aria-hidden="true" tabindex="-1"></a>      modt <span class="ot">=</span> <span class="fu">try</span>(<span class="fu">model.matrix</span>(lm.fm<span class="sc">$</span>lm.formula, <span class="at">data=</span>ftemp, <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>lm.fm<span class="sc">$</span>lm.params))), <span class="at">silent=</span><span class="cn">TRUE</span>)</span>
<span id="cb1-445"><a href="#cb1-445" aria-hidden="true" tabindex="-1"></a>      <span class="fu">options</span>(<span class="at">warn =</span> <span class="dv">0</span>)</span>
<span id="cb1-446"><a href="#cb1-446" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb1-447"><a href="#cb1-447" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span>(<span class="sc">!</span><span class="fu">inherits</span>(modt, <span class="st">&quot;try-error&quot;</span>)) { <span class="co"># do nothing if could not make model matrix</span></span>
<span id="cb1-448"><a href="#cb1-448" aria-hidden="true" tabindex="-1"></a>        <span class="fu">options</span>(<span class="at">warn =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb1-449"><a href="#cb1-449" aria-hidden="true" tabindex="-1"></a>        <span class="co"># if we are able to solve this, we are able to estimate bias  </span></span>
<span id="cb1-450"><a href="#cb1-450" aria-hidden="true" tabindex="-1"></a>        bhat <span class="ot">=</span>  <span class="fu">try</span>(<span class="fu">solve</span>(<span class="fu">t</span>(modt) <span class="sc">%*%</span> modt) <span class="sc">%*%</span> <span class="fu">t</span>(modt) <span class="sc">%*%</span> peptemp)</span>
<span id="cb1-451"><a href="#cb1-451" aria-hidden="true" tabindex="-1"></a>        <span class="fu">options</span>(<span class="at">warn =</span> <span class="dv">0</span>)</span>
<span id="cb1-452"><a href="#cb1-452" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>(<span class="sc">!</span><span class="fu">inherits</span>(bhat, <span class="st">&quot;try-error&quot;</span>)) {</span>
<span id="cb1-453"><a href="#cb1-453" aria-hidden="true" tabindex="-1"></a>          betahat[,ii] <span class="ot">=</span> bhat</span>
<span id="cb1-454"><a href="#cb1-454" aria-hidden="true" tabindex="-1"></a>          ceffects <span class="ot">=</span> modt <span class="sc">%*%</span> bhat  <span class="co"># these are the group effects, from estimated coefficients betahat</span></span>
<span id="cb1-455"><a href="#cb1-455" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb1-456"><a href="#cb1-456" aria-hidden="true" tabindex="-1"></a>          resm <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, numsamp) <span class="co"># really a vector only, not m </span></span>
<span id="cb1-457"><a href="#cb1-457" aria-hidden="true" tabindex="-1"></a>          resm[pos] <span class="ot">=</span> <span class="fu">as.numeric</span>(pep[pos] <span class="sc">-</span> ceffects)</span>
<span id="cb1-458"><a href="#cb1-458" aria-hidden="true" tabindex="-1"></a>          bias <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, numsamp)</span>
<span id="cb1-459"><a href="#cb1-459" aria-hidden="true" tabindex="-1"></a>          bias[pos] <span class="ot">=</span> resm[pos] <span class="sc">%*%</span> V0[pos,] <span class="sc">%*%</span> <span class="fu">t</span>(V0[pos,])</span>
<span id="cb1-460"><a href="#cb1-460" aria-hidden="true" tabindex="-1"></a>          norm_m[ii, ] <span class="ot">=</span> <span class="fu">as.numeric</span>(pep <span class="sc">-</span> bias)</span>
<span id="cb1-461"><a href="#cb1-461" aria-hidden="true" tabindex="-1"></a>          </span>
<span id="cb1-462"><a href="#cb1-462" aria-hidden="true" tabindex="-1"></a>          <span class="co"># yuliya:  but newR should be computed on Normalized data</span></span>
<span id="cb1-463"><a href="#cb1-463" aria-hidden="true" tabindex="-1"></a>          resm_n <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">NA</span>, numsamp)</span>
<span id="cb1-464"><a href="#cb1-464" aria-hidden="true" tabindex="-1"></a>          bhat_n <span class="ot">=</span>  <span class="fu">solve</span>(<span class="fu">t</span>(modt) <span class="sc">%*%</span> modt) <span class="sc">%*%</span> <span class="fu">t</span>(modt) <span class="sc">%*%</span> norm_m[ii, pos]</span>
<span id="cb1-465"><a href="#cb1-465" aria-hidden="true" tabindex="-1"></a>          betahat_n[,ii] <span class="ot">=</span> bhat_n</span>
<span id="cb1-466"><a href="#cb1-466" aria-hidden="true" tabindex="-1"></a>          ceffects_n <span class="ot">=</span> modt <span class="sc">%*%</span> bhat_n</span>
<span id="cb1-467"><a href="#cb1-467" aria-hidden="true" tabindex="-1"></a>          resm_n[pos] <span class="ot">=</span> norm_m[ii,pos] <span class="sc">-</span> ceffects</span>
<span id="cb1-468"><a href="#cb1-468" aria-hidden="true" tabindex="-1"></a>          newR[ii, ] <span class="ot">=</span> resm_n</span>
<span id="cb1-469"><a href="#cb1-469" aria-hidden="true" tabindex="-1"></a>        } <span class="cf">else</span> {</span>
<span id="cb1-470"><a href="#cb1-470" aria-hidden="true" tabindex="-1"></a>          <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;got exception 2 at peptide:&#39;</span>, ii, <span class="st">&#39;should not get here...&#39;</span>)) </span>
<span id="cb1-471"><a href="#cb1-471" aria-hidden="true" tabindex="-1"></a>          exPeps[ii] <span class="ot">=</span> <span class="dv">2</span> <span class="co"># should not get 2 here ever...</span></span>
<span id="cb1-472"><a href="#cb1-472" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb1-473"><a href="#cb1-473" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb1-474"><a href="#cb1-474" aria-hidden="true" tabindex="-1"></a>        <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;got exception at peptide:&#39;</span>, ii)) </span>
<span id="cb1-475"><a href="#cb1-475" aria-hidden="true" tabindex="-1"></a>        exPeps[ii] <span class="ot">=</span> <span class="dv">1</span> <span class="co"># keep track of peptides that threw exeptions, check why...</span></span>
<span id="cb1-476"><a href="#cb1-476" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb1-477"><a href="#cb1-477" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-478"><a href="#cb1-478" aria-hidden="true" tabindex="-1"></a>  } <span class="co"># end else - got 2+ treatment groups</span></span>
<span id="cb1-479"><a href="#cb1-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-480"><a href="#cb1-480" aria-hidden="true" tabindex="-1"></a>  <span class="do">#####################################################################################</span></span>
<span id="cb1-481"><a href="#cb1-481" aria-hidden="true" tabindex="-1"></a>  <span class="co"># rescaling has been eliminated form the code after discussion that bias </span></span>
<span id="cb1-482"><a href="#cb1-482" aria-hidden="true" tabindex="-1"></a>  <span class="co"># adds variation and we remove it, so no need to rescale after as we removed what was introduced</span></span>
<span id="cb1-483"><a href="#cb1-483" aria-hidden="true" tabindex="-1"></a>  y_rescaled <span class="ot">=</span> norm_m <span class="co"># for 1 group normalization only, we do not rescale</span></span>
<span id="cb1-484"><a href="#cb1-484" aria-hidden="true" tabindex="-1"></a>  <span class="co"># add column names to y-rescaled, now X1, X2,...</span></span>
<span id="cb1-485"><a href="#cb1-485" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(y_rescaled) <span class="ot">=</span> <span class="fu">colnames</span>(pres) <span class="co"># these have same number of cols</span></span>
<span id="cb1-486"><a href="#cb1-486" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(y_rescaled) <span class="ot">=</span> <span class="fu">rownames</span>(pres) </span>
<span id="cb1-487"><a href="#cb1-487" aria-hidden="true" tabindex="-1"></a>  y_resc <span class="ot">=</span> <span class="fu">data.frame</span>(present, y_rescaled)  </span>
<span id="cb1-488"><a href="#cb1-488" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(y_resc) <span class="ot">=</span> <span class="fu">rownames</span>(pres)  <span class="co"># rownames(rv$normalized)</span></span>
<span id="cb1-489"><a href="#cb1-489" aria-hidden="true" tabindex="-1"></a>  final <span class="ot">=</span> y_resc <span class="co"># row names are assumed to be UNIQUE, peptide IDs are unique</span></span>
<span id="cb1-490"><a href="#cb1-490" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-491"><a href="#cb1-491" aria-hidden="true" tabindex="-1"></a>  <span class="co"># rows with all observations present</span></span>
<span id="cb1-492"><a href="#cb1-492" aria-hidden="true" tabindex="-1"></a>  complete_all <span class="ot">=</span> y_rescaled[<span class="fu">rowSums</span>(<span class="fu">is.na</span>(y_rescaled))<span class="sc">==</span><span class="dv">0</span>,,drop<span class="ot">=</span>F]</span>
<span id="cb1-493"><a href="#cb1-493" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-494"><a href="#cb1-494" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  x11() # make R open new figure window</span></span>
<span id="cb1-495"><a href="#cb1-495" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfcol=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb1-496"><a href="#cb1-496" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb1-497"><a href="#cb1-497" aria-hidden="true" tabindex="-1"></a>  <span class="co"># center each peptide around zero (subtract its mean across samples)</span></span>
<span id="cb1-498"><a href="#cb1-498" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note: we are not changing matrix itself, only centerig what we pass to svd</span></span>
<span id="cb1-499"><a href="#cb1-499" aria-hidden="true" tabindex="-1"></a>  complete_all_center <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(complete_all), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>))</span>
<span id="cb1-500"><a href="#cb1-500" aria-hidden="true" tabindex="-1"></a>  toplot3 <span class="ot">=</span> <span class="fu">svd</span>(complete_all_center)</span>
<span id="cb1-501"><a href="#cb1-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.eigentrends</span>(toplot1, <span class="st">&quot;Raw Data&quot;</span>)</span>
<span id="cb1-502"><a href="#cb1-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot.eigentrends</span>(toplot3, <span class="st">&quot;Normalized Data&quot;</span>)</span>
<span id="cb1-503"><a href="#cb1-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-504"><a href="#cb1-504" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;Done with normalization!!!&quot;</span>)</span>
<span id="cb1-505"><a href="#cb1-505" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(V0) <span class="ot">=</span>  <span class="fu">paste</span>(<span class="st">&quot;Trend&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(V0), <span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1-506"><a href="#cb1-506" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-507"><a href="#cb1-507" aria-hidden="true" tabindex="-1"></a>  maxrange <span class="ot">=</span> <span class="cn">NULL</span> <span class="co"># no rescaling # data.matrix(maxrange)</span></span>
<span id="cb1-508"><a href="#cb1-508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">normalized=</span>final, <span class="at">norm_m=</span>y_rescaled, <span class="at">eigentrends=</span>V0, <span class="at">rescrange=</span>maxrange, </span>
<span id="cb1-509"><a href="#cb1-509" aria-hidden="true" tabindex="-1"></a>              <span class="at">norm.svd=</span>toplot3, <span class="at">exPeps=</span>exPeps)) </span>
<span id="cb1-510"><a href="#cb1-510" aria-hidden="true" tabindex="-1"></a>} <span class="co"># end function eig_norm2</span></span>
<span id="cb1-511"><a href="#cb1-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-512"><a href="#cb1-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-513"><a href="#cb1-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-514"><a href="#cb1-514" aria-hidden="true" tabindex="-1"></a><span class="do">### EigenMS helper functions, a few more...</span></span>
<span id="cb1-515"><a href="#cb1-515" aria-hidden="true" tabindex="-1"></a><span class="co"># Tom had Sig set to 0.1, I and Storey&#39;s paper has 0.05</span></span>
<span id="cb1-516"><a href="#cb1-516" aria-hidden="true" tabindex="-1"></a><span class="co"># sva.id = function(dat, mod, n.u.treatment, B=500, sv.sig=0.05, seed=NULL) {</span></span>
<span id="cb1-517"><a href="#cb1-517" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-518"><a href="#cb1-518" aria-hidden="true" tabindex="-1"></a><span class="co"># yuliya Sept 1, 2014: changed parameter mod to be treatment, using lm() to obtain residuals</span></span>
<span id="cb1-519"><a href="#cb1-519" aria-hidden="true" tabindex="-1"></a>sva.id <span class="ot">=</span> <span class="cf">function</span>(dat, treatment, n.u.treatment, lm.fm, <span class="at">B=</span><span class="dv">500</span>, <span class="at">sv.sig=</span><span class="fl">0.05</span>, <span class="at">seed=</span><span class="cn">NULL</span>) {</span>
<span id="cb1-520"><a href="#cb1-520" aria-hidden="true" tabindex="-1"></a><span class="co"># Executes Surrogate Variable Analysis</span></span>
<span id="cb1-521"><a href="#cb1-521" aria-hidden="true" tabindex="-1"></a><span class="co"># Input:</span></span>
<span id="cb1-522"><a href="#cb1-522" aria-hidden="true" tabindex="-1"></a><span class="co">#   dat: A m peptides/genes by n samples matrix of expression data</span></span>
<span id="cb1-523"><a href="#cb1-523" aria-hidden="true" tabindex="-1"></a><span class="co">#   mod: A model matrix for the terms included in the analysis </span></span>
<span id="cb1-524"><a href="#cb1-524" aria-hidden="true" tabindex="-1"></a><span class="co">#   n.u.treatment - 0 or 1, if we are normalizing data with NO groups or some groups, QC vs samples</span></span>
<span id="cb1-525"><a href="#cb1-525" aria-hidden="true" tabindex="-1"></a><span class="co">#   B: The number of null iterations to perform</span></span>
<span id="cb1-526"><a href="#cb1-526" aria-hidden="true" tabindex="-1"></a><span class="co">#   sv.sig: The significance cutoff for the surrogate variables</span></span>
<span id="cb1-527"><a href="#cb1-527" aria-hidden="true" tabindex="-1"></a><span class="co">#   seed: A seed value for reproducible results</span></span>
<span id="cb1-528"><a href="#cb1-528" aria-hidden="true" tabindex="-1"></a><span class="co"># Output</span></span>
<span id="cb1-529"><a href="#cb1-529" aria-hidden="true" tabindex="-1"></a><span class="co">#    n.sv: Number of significant surrogate variables. </span></span>
<span id="cb1-530"><a href="#cb1-530" aria-hidden="true" tabindex="-1"></a><span class="co">#    id: An indicator of the significant surrogate variables</span></span>
<span id="cb1-531"><a href="#cb1-531" aria-hidden="true" tabindex="-1"></a><span class="co">#    B: number of permutation to do</span></span>
<span id="cb1-532"><a href="#cb1-532" aria-hidden="true" tabindex="-1"></a><span class="co">#    sv.sig: significance level for surrogate variables</span></span>
<span id="cb1-533"><a href="#cb1-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="st">&quot;Number of complete peptides (and samples) used in SVD&quot;</span>)</span>
<span id="cb1-534"><a href="#cb1-534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">dim</span>(dat))</span>
<span id="cb1-535"><a href="#cb1-535" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-536"><a href="#cb1-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-537"><a href="#cb1-537" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="fu">is.null</span>(seed))  { <span class="fu">set.seed</span>(seed) }</span>
<span id="cb1-538"><a href="#cb1-538" aria-hidden="true" tabindex="-1"></a>  warn <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb1-539"><a href="#cb1-539" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">=</span> <span class="fu">ncol</span>(dat)</span>
<span id="cb1-540"><a href="#cb1-540" aria-hidden="true" tabindex="-1"></a>  m <span class="ot">=</span> <span class="fu">nrow</span>(dat)</span>
<span id="cb1-541"><a href="#cb1-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-542"><a href="#cb1-542" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ncomp = length(as.numeric(n.u.treatment))</span></span>
<span id="cb1-543"><a href="#cb1-543" aria-hidden="true" tabindex="-1"></a>  ncomp <span class="ot">=</span> n.u.treatment <span class="co"># JULY 2013: as.numeric(n.u.treatment)</span></span>
<span id="cb1-544"><a href="#cb1-544" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&quot;Number of treatment groups (in svd.id): &quot;</span>, ncomp))</span>
<span id="cb1-545"><a href="#cb1-545" aria-hidden="true" tabindex="-1"></a>  <span class="co"># should be true for either case and can be used later</span></span>
<span id="cb1-546"><a href="#cb1-546" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-547"><a href="#cb1-547" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(ncomp <span class="sc">&gt;</span> <span class="dv">1</span>) { <span class="co">#   </span></span>
<span id="cb1-548"><a href="#cb1-548" aria-hidden="true" tabindex="-1"></a>    formula1 <span class="ot">=</span> <span class="fu">paste</span>(<span class="st">&#39;t(dat)~&#39;</span>, <span class="fu">as.character</span>(lm.fm<span class="sc">$</span>lm.formula)[<span class="dv">2</span>], <span class="at">sep =</span> <span class="st">&#39;&#39;</span>)</span>
<span id="cb1-549"><a href="#cb1-549" aria-hidden="true" tabindex="-1"></a>    fit_lmAll <span class="ot">=</span> <span class="fu">lm</span>(<span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text=</span>formula1)))</span>
<span id="cb1-550"><a href="#cb1-550" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">residuals</span>(fit_lmAll))</span>
<span id="cb1-551"><a href="#cb1-551" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb1-552"><a href="#cb1-552" aria-hidden="true" tabindex="-1"></a>    res <span class="ot">=</span> dat</span>
<span id="cb1-553"><a href="#cb1-553" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-554"><a href="#cb1-554" aria-hidden="true" tabindex="-1"></a>  <span class="co"># centering was not done before...</span></span>
<span id="cb1-555"><a href="#cb1-555" aria-hidden="true" tabindex="-1"></a>  <span class="co"># center each peptide around zero (subtract its mean across samples)</span></span>
<span id="cb1-556"><a href="#cb1-556" aria-hidden="true" tabindex="-1"></a>  <span class="co"># note: we are not changing matrix itself, only centerig what we pass to svd</span></span>
<span id="cb1-557"><a href="#cb1-557" aria-hidden="true" tabindex="-1"></a>  res_center <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(res), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>))</span>
<span id="cb1-558"><a href="#cb1-558" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-559"><a href="#cb1-559" aria-hidden="true" tabindex="-1"></a>  uu <span class="ot">=</span> <span class="fu">svd</span>(<span class="fu">t</span>(res_center)) <span class="co"># NEED a WRAPPER for t(). the diag is min(n, m)</span></span>
<span id="cb1-560"><a href="#cb1-560" aria-hidden="true" tabindex="-1"></a>  temp <span class="ot">=</span> uu<span class="sc">$</span>u</span>
<span id="cb1-561"><a href="#cb1-561" aria-hidden="true" tabindex="-1"></a>  uu<span class="sc">$</span>u <span class="ot">=</span> uu<span class="sc">$</span>v</span>
<span id="cb1-562"><a href="#cb1-562" aria-hidden="true" tabindex="-1"></a>  uu<span class="sc">$</span>v <span class="ot">=</span> temp</span>
<span id="cb1-563"><a href="#cb1-563" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-564"><a href="#cb1-564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-565"><a href="#cb1-565" aria-hidden="true" tabindex="-1"></a>  <span class="co"># yuliya: Sept 2014: can I get around without using H?? </span></span>
<span id="cb1-566"><a href="#cb1-566" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  ndf = min(n, m) - ceiling(sum(diag(H)))  </span></span>
<span id="cb1-567"><a href="#cb1-567" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  dstat = uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)</span></span>
<span id="cb1-568"><a href="#cb1-568" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  dstat0 = matrix(0,nrow=B,ncol=ndf)</span></span>
<span id="cb1-569"><a href="#cb1-569" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  s0 = diag(uu$d) # no need for diag here, should investigate why this is a vector already...</span></span>
<span id="cb1-570"><a href="#cb1-570" aria-hidden="true" tabindex="-1"></a>  s0 <span class="ot">=</span> uu<span class="sc">$</span>d</span>
<span id="cb1-571"><a href="#cb1-571" aria-hidden="true" tabindex="-1"></a>  s0 <span class="ot">=</span> s0<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb1-572"><a href="#cb1-572" aria-hidden="true" tabindex="-1"></a>  dstat <span class="ot">=</span> s0<span class="sc">/</span><span class="fu">sum</span>(s0)  <span class="co"># this did not have &#39;diag&#39; in it in Tom&#39;s code...</span></span>
<span id="cb1-573"><a href="#cb1-573" aria-hidden="true" tabindex="-1"></a>  ndf <span class="ot">=</span> <span class="fu">length</span>(dstat) <span class="co"># sticking to Tom&#39;s variable name</span></span>
<span id="cb1-574"><a href="#cb1-574" aria-hidden="true" tabindex="-1"></a><span class="co"># print(paste(&#39;length(dstat) = ndf = &#39;, ndf))</span></span>
<span id="cb1-575"><a href="#cb1-575" aria-hidden="true" tabindex="-1"></a>  dstat0 <span class="ot">=</span> <span class="fu">matrix</span>(<span class="dv">0</span>,<span class="at">nrow=</span>B,<span class="at">ncol=</span>ndf) <span class="co"># num samples (?)</span></span>
<span id="cb1-576"><a href="#cb1-576" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-577"><a href="#cb1-577" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="st">&quot;Starting Bootstrap.....&quot;</span>)</span>
<span id="cb1-578"><a href="#cb1-578" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the Bootstrap procedure that determines the number of significant eigertrends... </span></span>
<span id="cb1-579"><a href="#cb1-579" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>B){</span>
<span id="cb1-580"><a href="#cb1-580" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(ii <span class="sc">%%</span> <span class="dv">50</span> <span class="sc">==</span> <span class="dv">0</span>) { <span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">&#39;Iteration &#39;</span>, ii)) }</span>
<span id="cb1-581"><a href="#cb1-581" aria-hidden="true" tabindex="-1"></a>    res0 <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">apply</span>(res, <span class="dv">1</span>, sample, <span class="at">replace=</span><span class="cn">FALSE</span>)) <span class="co"># regression</span></span>
<span id="cb1-582"><a href="#cb1-582" aria-hidden="true" tabindex="-1"></a>    <span class="co"># yuliya: not sure if this is needed at all</span></span>
<span id="cb1-583"><a href="#cb1-583" aria-hidden="true" tabindex="-1"></a>    <span class="co"># not needed for 1 group normalizaiton</span></span>
<span id="cb1-584"><a href="#cb1-584" aria-hidden="true" tabindex="-1"></a>    <span class="do">##### res0 = res0 - t(H %*% t(res0))</span></span>
<span id="cb1-585"><a href="#cb1-585" aria-hidden="true" tabindex="-1"></a>    <span class="co"># yuliya: Sept 3, 2014: REMOVED above line. Do not think this needs to be done.. </span></span>
<span id="cb1-586"><a href="#cb1-586" aria-hidden="true" tabindex="-1"></a>    <span class="co"># center each peptide around zero (subtract its mean across samples)</span></span>
<span id="cb1-587"><a href="#cb1-587" aria-hidden="true" tabindex="-1"></a>    <span class="co"># note: we are not changing matrix itself, only centerig what we pass to svd</span></span>
<span id="cb1-588"><a href="#cb1-588" aria-hidden="true" tabindex="-1"></a>    res0_center <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(res0), <span class="at">center =</span> <span class="cn">TRUE</span>, <span class="at">scale =</span> <span class="cn">FALSE</span>))</span>
<span id="cb1-589"><a href="#cb1-589" aria-hidden="true" tabindex="-1"></a>    uu0 <span class="ot">=</span> <span class="fu">svd</span>(res0_center)</span>
<span id="cb1-590"><a href="#cb1-590" aria-hidden="true" tabindex="-1"></a>    temp <span class="ot">=</span> uu0<span class="sc">$</span>u  <span class="co"># why did tom do this??</span></span>
<span id="cb1-591"><a href="#cb1-591" aria-hidden="true" tabindex="-1"></a>    uu0<span class="sc">$</span>u <span class="ot">=</span> uu0<span class="sc">$</span>v</span>
<span id="cb1-592"><a href="#cb1-592" aria-hidden="true" tabindex="-1"></a>    uu0<span class="sc">$</span>v <span class="ot">=</span> temp</span>
<span id="cb1-593"><a href="#cb1-593" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-594"><a href="#cb1-594" aria-hidden="true" tabindex="-1"></a>    ss0 <span class="ot">=</span> uu0<span class="sc">$</span>d  <span class="co"># no need for diag.... </span></span>
<span id="cb1-595"><a href="#cb1-595" aria-hidden="true" tabindex="-1"></a>    ss0 <span class="ot">=</span> ss0<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb1-596"><a href="#cb1-596" aria-hidden="true" tabindex="-1"></a>    dstat0[ii,] <span class="ot">=</span> ss0 <span class="sc">/</span> <span class="fu">sum</span>(ss0) <span class="co"># Tk0 in Matlab</span></span>
<span id="cb1-597"><a href="#cb1-597" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-598"><a href="#cb1-598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-599"><a href="#cb1-599" aria-hidden="true" tabindex="-1"></a><span class="co"># yuliya: check p-values here, Tom had mean value...</span></span>
<span id="cb1-600"><a href="#cb1-600" aria-hidden="true" tabindex="-1"></a>  psv <span class="ot">=</span> <span class="fu">rep</span>(<span class="dv">1</span>,n)</span>
<span id="cb1-601"><a href="#cb1-601" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>ndf){</span>
<span id="cb1-602"><a href="#cb1-602" aria-hidden="true" tabindex="-1"></a>      <span class="co"># psv[ii] = mean(dstat0[,ii] &gt;= dstat[ii])</span></span>
<span id="cb1-603"><a href="#cb1-603" aria-hidden="true" tabindex="-1"></a>    <span class="co"># should this be compared to a MEAN?  Should this be dstat0[ii,] ?</span></span>
<span id="cb1-604"><a href="#cb1-604" aria-hidden="true" tabindex="-1"></a>    posGreater <span class="ot">=</span> dstat0[,ii] <span class="sc">&gt;</span> dstat[ii]</span>
<span id="cb1-605"><a href="#cb1-605" aria-hidden="true" tabindex="-1"></a>    psv[ii] <span class="ot">=</span> <span class="fu">sum</span>(posGreater) <span class="sc">/</span> B</span>
<span id="cb1-606"><a href="#cb1-606" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-607"><a href="#cb1-607" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-608"><a href="#cb1-608" aria-hidden="true" tabindex="-1"></a>  <span class="co"># p-values for peptides have to be in monotonically increasing order, </span></span>
<span id="cb1-609"><a href="#cb1-609" aria-hidden="true" tabindex="-1"></a>  <span class="co"># set equal to previous one if not the case</span></span>
<span id="cb1-610"><a href="#cb1-610" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(ii <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span>ndf){</span>
<span id="cb1-611"><a href="#cb1-611" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(psv[(ii<span class="dv">-1</span>)] <span class="sc">&gt;</span> psv[ii]) {</span>
<span id="cb1-612"><a href="#cb1-612" aria-hidden="true" tabindex="-1"></a>        <span class="co"># psv[ii] = max(psv[(ii-1)],psv[ii]) </span></span>
<span id="cb1-613"><a href="#cb1-613" aria-hidden="true" tabindex="-1"></a>      psv[ii] <span class="ot">=</span> psv[(ii<span class="dv">-1</span>)] </span>
<span id="cb1-614"><a href="#cb1-614" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-615"><a href="#cb1-615" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-616"><a href="#cb1-616" aria-hidden="true" tabindex="-1"></a>  nsv <span class="ot">=</span> <span class="fu">sum</span>(psv <span class="sc">&lt;=</span> sv.sig)</span>
<span id="cb1-617"><a href="#cb1-617" aria-hidden="true" tabindex="-1"></a>  <span class="co"># tom - this should be at least 1</span></span>
<span id="cb1-618"><a href="#cb1-618" aria-hidden="true" tabindex="-1"></a>  <span class="co"># nsv = min(sum(psv &lt;= sv.sig), 1, na.rm=T)</span></span>
<span id="cb1-619"><a href="#cb1-619" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">n.sv =</span> nsv,<span class="at">p.sv=</span>psv))</span>
<span id="cb1-620"><a href="#cb1-620" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-621"><a href="#cb1-621" aria-hidden="true" tabindex="-1"></a><span class="co"># end sva.id</span></span>
<span id="cb1-622"><a href="#cb1-622" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb1-623"><a href="#cb1-623" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-624"><a href="#cb1-624" aria-hidden="true" tabindex="-1"></a>mmul <span class="ot">=</span> <span class="cf">function</span>(A, B){</span>
<span id="cb1-625"><a href="#cb1-625" aria-hidden="true" tabindex="-1"></a>  <span class="co"># multiply square matrix by rectangle with NA&#39;s (???)</span></span>
<span id="cb1-626"><a href="#cb1-626" aria-hidden="true" tabindex="-1"></a>  X <span class="ot">=</span> A</span>
<span id="cb1-627"><a href="#cb1-627" aria-hidden="true" tabindex="-1"></a>  Y <span class="ot">=</span> B</span>
<span id="cb1-628"><a href="#cb1-628" aria-hidden="true" tabindex="-1"></a>  X[<span class="fu">is.na</span>(A)] <span class="ot">=</span> Y[<span class="fu">is.na</span>(B)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb1-629"><a href="#cb1-629" aria-hidden="true" tabindex="-1"></a>  R <span class="ot">=</span> X <span class="sc">%*%</span> Y</span>
<span id="cb1-630"><a href="#cb1-630" aria-hidden="true" tabindex="-1"></a>  R[<span class="fu">is.na</span>(B)] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb1-631"><a href="#cb1-631" aria-hidden="true" tabindex="-1"></a>  R</span>
<span id="cb1-632"><a href="#cb1-632" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-633"><a href="#cb1-633" aria-hidden="true" tabindex="-1"></a><span class="co"># end mmul</span></span>
<span id="cb1-634"><a href="#cb1-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-635"><a href="#cb1-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-636"><a href="#cb1-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-637"><a href="#cb1-637" aria-hidden="true" tabindex="-1"></a><span class="do">#####    Not used in EigenMS, but a solution to imputing peptides with 1+ group missing completely.</span></span>
<span id="cb1-638"><a href="#cb1-638" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb1-639"><a href="#cb1-639" aria-hidden="true" tabindex="-1"></a>imp_elim_proteins<span class="ot">=</span> <span class="cf">function</span>(nosib.miss, treatment){</span>
<span id="cb1-640"><a href="#cb1-640" aria-hidden="true" tabindex="-1"></a><span class="co"># Impute 1 peptide proteins with 1 grp completely missing</span></span>
<span id="cb1-641"><a href="#cb1-641" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb1-642"><a href="#cb1-642" aria-hidden="true" tabindex="-1"></a><span class="co"># INPUT: proteins - 1 peptide proteins with 1 grp missing completely (NaNs)</span></span>
<span id="cb1-643"><a href="#cb1-643" aria-hidden="true" tabindex="-1"></a><span class="co">#        grps - grouping information for observations, can be 2+ grps here, </span></span>
<span id="cb1-644"><a href="#cb1-644" aria-hidden="true" tabindex="-1"></a><span class="co">#               but not sure what to do with more then 2 groups if 2+ gs are</span></span>
<span id="cb1-645"><a href="#cb1-645" aria-hidden="true" tabindex="-1"></a><span class="co">#               missing completely. So only 1 grp is completely missing.</span></span>
<span id="cb1-646"><a href="#cb1-646" aria-hidden="true" tabindex="-1"></a><span class="co"># OUTPUT: prs - imputed proteins, same size as proteins parameter</span></span>
<span id="cb1-647"><a href="#cb1-647" aria-hidden="true" tabindex="-1"></a><span class="co">#nosib.miss &lt;= nosib.miss</span></span>
<span id="cb1-648"><a href="#cb1-648" aria-hidden="true" tabindex="-1"></a><span class="co">#treatment &lt;= treatment </span></span>
<span id="cb1-649"><a href="#cb1-649" aria-hidden="true" tabindex="-1"></a>  tlvls <span class="ot">=</span> <span class="fu">unique</span>(treatment)</span>
<span id="cb1-650"><a href="#cb1-650" aria-hidden="true" tabindex="-1"></a>  proteins <span class="ot">=</span> nosib.miss</span>
<span id="cb1-651"><a href="#cb1-651" aria-hidden="true" tabindex="-1"></a>  ng <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">unique</span>(treatment))</span>
<span id="cb1-652"><a href="#cb1-652" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(proteins)) {</span>
<span id="cb1-653"><a href="#cb1-653" aria-hidden="true" tabindex="-1"></a>    pr <span class="ot">=</span> <span class="fu">as.vector</span>(proteins[i,])</span>
<span id="cb1-654"><a href="#cb1-654" aria-hidden="true" tabindex="-1"></a>    <span class="co">#% find number of missing values in ALL groups</span></span>
<span id="cb1-655"><a href="#cb1-655" aria-hidden="true" tabindex="-1"></a>    miss <span class="ot">=</span> <span class="fu">array</span>(<span class="cn">NA</span>, <span class="fu">c</span>(<span class="dv">1</span>, ng))</span>
<span id="cb1-656"><a href="#cb1-656" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>ng) miss[j] <span class="ot">=</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(pr[treatment<span class="sc">==</span><span class="fu">unique</span>(treatment)[j]]))</span>
<span id="cb1-657"><a href="#cb1-657" aria-hidden="true" tabindex="-1"></a><span class="co">#    pos = miss==0</span></span>
<span id="cb1-658"><a href="#cb1-658" aria-hidden="true" tabindex="-1"></a>    pos <span class="ot">=</span> miss<span class="sc">==</span><span class="fu">min</span>(miss)  <span class="co"># Tom 092510</span></span>
<span id="cb1-659"><a href="#cb1-659" aria-hidden="true" tabindex="-1"></a>    present_groups <span class="ot">=</span> pr[treatment<span class="sc">==</span><span class="fu">unique</span>(treatment)[pos]]</span>
<span id="cb1-660"><a href="#cb1-660" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-661"><a href="#cb1-661" aria-hidden="true" tabindex="-1"></a>    <span class="co"># compute mean and stdev from one of the present groups, make sure no NaNs are used</span></span>
<span id="cb1-662"><a href="#cb1-662" aria-hidden="true" tabindex="-1"></a>    pospres <span class="ot">=</span> <span class="sc">!</span><span class="fu">is.na</span>(present_groups)</span>
<span id="cb1-663"><a href="#cb1-663" aria-hidden="true" tabindex="-1"></a>    presvals <span class="ot">=</span> present_groups[pospres]</span>
<span id="cb1-664"><a href="#cb1-664" aria-hidden="true" tabindex="-1"></a>    pepmean <span class="ot">=</span> <span class="fu">mean</span>(presvals)</span>
<span id="cb1-665"><a href="#cb1-665" aria-hidden="true" tabindex="-1"></a>    pepstd <span class="ot">=</span>  <span class="fu">sd</span>(presvals)  </span>
<span id="cb1-666"><a href="#cb1-666" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">is.na</span>(pepstd)) <span class="cf">next</span>;</span>
<span id="cb1-667"><a href="#cb1-667" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb1-668"><a href="#cb1-668" aria-hidden="true" tabindex="-1"></a>  <span class="co">#% imputing only COMPLETELY missing peptides here</span></span>
<span id="cb1-669"><a href="#cb1-669" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>ng) {</span>
<span id="cb1-670"><a href="#cb1-670" aria-hidden="true" tabindex="-1"></a>        <span class="cf">if</span>   (<span class="sc">!</span>pos[j]) { <span class="co">#% imute only the ones not at pos complete</span></span>
<span id="cb1-671"><a href="#cb1-671" aria-hidden="true" tabindex="-1"></a>            imppos <span class="ot">=</span> <span class="fu">is.na</span>(pr[treatment<span class="sc">==</span>tlvls[j]])  <span class="co">#% should be all in this group, but double check</span></span>
<span id="cb1-672"><a href="#cb1-672" aria-hidden="true" tabindex="-1"></a>            imppepmean <span class="ot">=</span> pepmean <span class="sc">-</span> <span class="dv">6</span><span class="sc">*</span> pepstd</span>
<span id="cb1-673"><a href="#cb1-673" aria-hidden="true" tabindex="-1"></a>            imppepstd <span class="ot">=</span> pepstd</span>
<span id="cb1-674"><a href="#cb1-674" aria-hidden="true" tabindex="-1"></a>            tt <span class="ot">=</span> imppepmean <span class="sc">-</span> <span class="dv">3</span> <span class="sc">*</span> imppepstd </span>
<span id="cb1-675"><a href="#cb1-675" aria-hidden="true" tabindex="-1"></a>            kk <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb1-676"><a href="#cb1-676" aria-hidden="true" tabindex="-1"></a>      <span class="cf">while</span> (tt <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;&amp;</span> kk <span class="sc">&lt;</span> <span class="dv">10</span>){  <span class="co"># added kk counter - tom 092510</span></span>
<span id="cb1-677"><a href="#cb1-677" aria-hidden="true" tabindex="-1"></a>                offset <span class="ot">=</span> .<span class="dv">25</span></span>
<span id="cb1-678"><a href="#cb1-678" aria-hidden="true" tabindex="-1"></a>                gap <span class="ot">=</span> imppepstd <span class="sc">*</span> offset</span>
<span id="cb1-679"><a href="#cb1-679" aria-hidden="true" tabindex="-1"></a>                imppepstd <span class="ot">=</span> imppepstd <span class="sc">*</span> (<span class="dv">1</span><span class="sc">-</span> offset)</span>
<span id="cb1-680"><a href="#cb1-680" aria-hidden="true" tabindex="-1"></a>                imppepmean <span class="ot">=</span> imppepmean <span class="sc">+</span> <span class="dv">3</span> <span class="sc">*</span> gap</span>
<span id="cb1-681"><a href="#cb1-681" aria-hidden="true" tabindex="-1"></a>                tt <span class="ot">=</span> imppepmean <span class="sc">-</span> <span class="dv">3</span> <span class="sc">*</span> imppepstd</span>
<span id="cb1-682"><a href="#cb1-682" aria-hidden="true" tabindex="-1"></a>                kk <span class="ot">=</span> kk <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-683"><a href="#cb1-683" aria-hidden="true" tabindex="-1"></a>            }</span>
<span id="cb1-684"><a href="#cb1-684" aria-hidden="true" tabindex="-1"></a>            imp_tmp <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(imppos), imppepmean, pepstd)</span>
<span id="cb1-685"><a href="#cb1-685" aria-hidden="true" tabindex="-1"></a>            pr[treatment<span class="sc">==</span>tlvls[j]] <span class="ot">=</span> imp_tmp</span>
<span id="cb1-686"><a href="#cb1-686" aria-hidden="true" tabindex="-1"></a>        }</span>
<span id="cb1-687"><a href="#cb1-687" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-688"><a href="#cb1-688" aria-hidden="true" tabindex="-1"></a>  proteins[i,] <span class="ot">=</span> pr </span>
<span id="cb1-689"><a href="#cb1-689" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-690"><a href="#cb1-690" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-691"><a href="#cb1-691" aria-hidden="true" tabindex="-1"></a>  <span class="co"># tom - this routine gives some nearly-blank rows</span></span>
<span id="cb1-692"><a href="#cb1-692" aria-hidden="true" tabindex="-1"></a>  <span class="co"># to avoid singularities, i&#39;m going to scan this to see which protein rows</span></span>
<span id="cb1-693"><a href="#cb1-693" aria-hidden="true" tabindex="-1"></a>  <span class="co"># have all blanks in one row and remove them</span></span>
<span id="cb1-694"><a href="#cb1-694" aria-hidden="true" tabindex="-1"></a>  <span class="co">#  proteins &lt;= proteins</span></span>
<span id="cb1-695"><a href="#cb1-695" aria-hidden="true" tabindex="-1"></a>  xx <span class="ot">=</span> (<span class="sc">!</span><span class="fu">is.na</span>(proteins)) <span class="sc">%*%</span> <span class="fu">model.matrix</span>(<span class="sc">~</span>treatment<span class="dv">-1</span>)</span>
<span id="cb1-696"><a href="#cb1-696" aria-hidden="true" tabindex="-1"></a>  notblank.idx <span class="ot">=</span> <span class="fu">rep</span>(<span class="cn">TRUE</span>, <span class="fu">nrow</span>(xx))</span>
<span id="cb1-697"><a href="#cb1-697" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(jj <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(xx)) notblank.idx <span class="ot">=</span> notblank.idx <span class="sc">&amp;</span> xx[,jj]</span>
<span id="cb1-698"><a href="#cb1-698" aria-hidden="true" tabindex="-1"></a><span class="co">#  proteins = proteins[blank.idx,,drop=FALSE]</span></span>
<span id="cb1-699"><a href="#cb1-699" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-700"><a href="#cb1-700" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">proteins=</span>proteins, <span class="at">notblank.idx=</span>notblank.idx))</span>
<span id="cb1-701"><a href="#cb1-701" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-702"><a href="#cb1-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-703"><a href="#cb1-703" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb1-704"><a href="#cb1-704" aria-hidden="true" tabindex="-1"></a><span class="do">#######################################################################</span></span>
<span id="cb1-705"><a href="#cb1-705" aria-hidden="true" tabindex="-1"></a>my.Psi <span class="ot">=</span> <span class="cf">function</span>(x, my.pi){</span>
<span id="cb1-706"><a href="#cb1-706" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates Psi</span></span>
<span id="cb1-707"><a href="#cb1-707" aria-hidden="true" tabindex="-1"></a><span class="fu">exp</span>(<span class="fu">log</span>(<span class="dv">1</span><span class="sc">-</span>my.pi)  <span class="sc">+</span> <span class="fu">dnorm</span>(x, <span class="dv">0</span>, <span class="dv">1</span>, <span class="at">log=</span>T) <span class="sc">-</span> <span class="fu">log</span>(my.pi <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> my.pi) <span class="sc">*</span> <span class="fu">pnorm</span>(x, <span class="dv">0</span>, <span class="dv">1</span>) ))</span>
<span id="cb1-708"><a href="#cb1-708" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-709"><a href="#cb1-709" aria-hidden="true" tabindex="-1"></a><span class="co"># end my.Psi</span></span>
<span id="cb1-710"><a href="#cb1-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-711"><a href="#cb1-711" aria-hidden="true" tabindex="-1"></a>my.Psi.dash <span class="ot">=</span> <span class="cf">function</span>(x, my.pi){</span>
<span id="cb1-712"><a href="#cb1-712" aria-hidden="true" tabindex="-1"></a><span class="co"># calculates the derivative of Psi</span></span>
<span id="cb1-713"><a href="#cb1-713" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span><span class="fu">my.Psi</span>(x, my.pi) <span class="sc">*</span> (x <span class="sc">+</span> <span class="fu">my.Psi</span>(x, my.pi))</span>
<span id="cb1-714"><a href="#cb1-714" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-715"><a href="#cb1-715" aria-hidden="true" tabindex="-1"></a><span class="co"># end my.Psi.dash</span></span>
<span id="cb1-716"><a href="#cb1-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-717"><a href="#cb1-717" aria-hidden="true" tabindex="-1"></a>phi <span class="ot">=</span> <span class="cf">function</span>(x){<span class="fu">dnorm</span>(x)}</span>
<span id="cb1-718"><a href="#cb1-718" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-719"><a href="#cb1-719" aria-hidden="true" tabindex="-1"></a>rnorm.trunc <span class="ot">=</span> <span class="cf">function</span> (n, mu, sigma, <span class="at">lo=</span><span class="sc">-</span><span class="cn">Inf</span>, <span class="at">hi=</span><span class="cn">Inf</span>){</span>
<span id="cb1-720"><a href="#cb1-720" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculates truncated noraml</span></span>
<span id="cb1-721"><a href="#cb1-721" aria-hidden="true" tabindex="-1"></a>  p.lo <span class="ot">=</span> <span class="fu">pnorm</span> (lo, mu, sigma)</span>
<span id="cb1-722"><a href="#cb1-722" aria-hidden="true" tabindex="-1"></a>  p.hi <span class="ot">=</span> <span class="fu">pnorm</span> (hi, mu, sigma)</span>
<span id="cb1-723"><a href="#cb1-723" aria-hidden="true" tabindex="-1"></a>  u <span class="ot">=</span> <span class="fu">runif</span> (n, p.lo, p.hi)</span>
<span id="cb1-724"><a href="#cb1-724" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span> (<span class="fu">qnorm</span> (u, mu, sigma))</span>
<span id="cb1-725"><a href="#cb1-725" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-726"><a href="#cb1-726" aria-hidden="true" tabindex="-1"></a><span class="co"># end rnorm.truncf</span></span></code></pre></div>
<div id="libraries" class="section level1">
<h1>Libraries</h1>
</div>
<div id="loading-the-data" class="section level1">
<h1>Loading the data</h1>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>WD <span class="ot">&lt;-</span> <span class="fu">getwd</span>()</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(WD<span class="sc">!=</span><span class="st">&quot;C:/Users/mysit/Documents/Bioinformatica/Semestre_4/workflowr/data&quot;</span>){</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">setwd</span>(<span class="st">&quot;../workflowr/data&quot;</span>)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>mdf <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="at">file =</span> <span class="st">&quot;mdf_processed.rds&quot;</span>)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>mdf<span class="ot">&lt;-</span>mdf <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">rowid =</span> <span class="fu">row_number</span>()) <span class="sc">%&gt;%</span> <span class="fu">relocate</span>(rowid, <span class="at">.before =</span> sample)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>rowinfo <span class="ot">&lt;-</span> mdf <span class="sc">%&gt;%</span> <span class="fu">select</span>(rowid, sample, age, batch, type, sample_id)</span></code></pre></div>
</div>
<div id="trying-to-make-it-work" class="section level1">
<h1>Trying to make it work</h1>
<div id="converting-to-log" class="section level2">
<h2>Converting to LOG</h2>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>raw <span class="ot">&lt;-</span> mdf <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;M&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">log2</span>()</span></code></pre></div>
</div>
<div id="defining-parameter-prot.info.-column-data-frame-with-ids-for-metabolites-or-peptides-in-case-of-matabolites-the-2-columns-are-identical." class="section level2">
<h2>Defining parameter prot.info. column data frame with IDs for metabolites or peptides in case of matabolites the 2 columns are identical.</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>m_prot.info <span class="ot">=</span> mdf[,<span class="fu">c</span>(<span class="dv">6</span>,<span class="dv">2</span>)]</span></code></pre></div>
</div>
<div id="defining-the-groups" class="section level2">
<h2>Defining the groups</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>grps <span class="ot">=</span> <span class="fu">as.factor</span>(<span class="fu">unique</span>(mdf<span class="sc">$</span>age))</span></code></pre></div>
</div>
<div id="first-step" class="section level2">
<h2>First step</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#m_ints_eig1 = eig_norm1(m=raw,treatment = grps,prot.info = m_prot.info)</span></span></code></pre></div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Catalan_Spain.1252  LC_CTYPE=Catalan_Spain.1252   
[3] LC_MONETARY=Catalan_Spain.1252 LC_NUMERIC=C                  
[5] LC_TIME=Catalan_Spain.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.4.0     patchwork_1.1.1  ggrepel_0.9.1    purrr_0.3.4     
 [5] dplyr_1.0.8      tidyr_1.2.0      readr_2.1.2      caret_6.0-91    
 [9] lattice_0.20-45  cowplot_1.1.1    plotly_4.10.0    ggfortify_0.4.14
[13] ggplot2_3.3.5   

loaded via a namespace (and not attached):
 [1] nlme_3.1-155         fs_1.5.2             lubridate_1.8.0     
 [4] httr_1.4.2           rprojroot_2.0.2      tools_4.1.3         
 [7] bslib_0.3.1          utf8_1.2.2           R6_2.5.1            
[10] rpart_4.1.16         DBI_1.1.2            lazyeval_0.2.2      
[13] colorspace_2.0-3     nnet_7.3-17          withr_2.5.0         
[16] tidyselect_1.1.2     gridExtra_2.3        compiler_4.1.3      
[19] git2r_0.30.1         cli_3.2.0            sass_0.4.1          
[22] scales_1.1.1         stringr_1.4.0        digest_0.6.29       
[25] rmarkdown_2.13       pkgconfig_2.0.3      htmltools_0.5.2     
[28] parallelly_1.30.0    fastmap_1.1.0        highr_0.9           
[31] htmlwidgets_1.5.4    rlang_1.0.2          rstudioapi_0.13     
[34] jquerylib_0.1.4      generics_0.1.2       jsonlite_1.8.0      
[37] ModelMetrics_1.2.2.2 magrittr_2.0.2       Matrix_1.4-0        
[40] Rcpp_1.0.8.3         munsell_0.5.0        fansi_1.0.3         
[43] lifecycle_1.0.1      stringi_1.7.6        whisker_0.4         
[46] pROC_1.18.0          yaml_2.3.5           MASS_7.3-55         
[49] plyr_1.8.7           recipes_0.2.0        grid_4.1.3          
[52] parallel_4.1.3       listenv_0.8.0        promises_1.2.0.1    
[55] crayon_1.5.1         splines_4.1.3        hms_1.1.1           
[58] knitr_1.38           pillar_1.7.0         stats4_4.1.3        
[61] future.apply_1.8.1   reshape2_1.4.4       codetools_0.2-18    
[64] glue_1.6.2           packrat_0.7.0        evaluate_0.15       
[67] data.table_1.14.2    tzdb_0.3.0           vctrs_0.3.8         
[70] httpuv_1.6.5         foreach_1.5.2        cellranger_1.1.0    
[73] gtable_0.3.0         future_1.24.0        assertthat_0.2.1    
[76] xfun_0.30            gower_1.0.0          prodlim_2019.11.13  
[79] later_1.3.0          class_7.3-20         survival_3.2-13     
[82] viridisLite_0.4.0    timeDate_3043.102    tibble_3.1.6        
[85] iterators_1.0.14     hardhat_0.2.0        workflowr_1.7.0     
[88] lava_1.6.10          globals_0.14.0       ellipsis_0.3.2      
[91] ipred_0.9-12        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
